<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Lara Mangravite" />
  <meta name="author" content="Avery Sen" />
  <meta name="author" content="John T. Wilbanks" />
  <meta name="dcterms.date" content="2020-06-18" />
  <meta name="keywords" content="governance, data sharing, open science, open data" />
  <title>Mechanisms to Govern Responsible Sharing of Open Data: A Progress Report</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/master/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="Mechanisms to Govern Responsible Sharing of Open Data: A Progress Report" />
  <meta name="citation_title" content="Mechanisms to Govern Responsible Sharing of Open Data: A Progress Report" />
  <meta property="og:title" content="Mechanisms to Govern Responsible Sharing of Open Data: A Progress Report" />
  <meta property="twitter:title" content="Mechanisms to Govern Responsible Sharing of Open Data: A Progress Report" />
  <meta name="dc.date" content="2020-06-18" />
  <meta name="citation_publication_date" content="2020-06-18" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Lara Mangravite" />
  <meta name="citation_author_institution" content="Sage Bionetworks" />
  <meta name="citation_author_orcid" content="0000-0001-7841-3612" />
  <meta name="citation_author" content="Avery Sen" />
  <meta name="citation_author_institution" content="Sentripetal" />
  <meta name="citation_author" content="John T. Wilbanks" />
  <meta name="citation_author_institution" content="Sage Bionetworks" />
  <meta name="citation_author_orcid" content="0000-0002-4510-0385" />
  <meta name="twitter:creator" content="@wilbanks" />
  <link rel="canonical" href="https://Sage-Bionetworks.github.io/governanceGreenPaper/" />
  <meta property="og:url" content="https://Sage-Bionetworks.github.io/governanceGreenPaper/" />
  <meta property="twitter:url" content="https://Sage-Bionetworks.github.io/governanceGreenPaper/" />
  <meta name="citation_fulltext_html_url" content="https://Sage-Bionetworks.github.io/governanceGreenPaper/" />
  <meta name="citation_pdf_url" content="https://Sage-Bionetworks.github.io/governanceGreenPaper/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://Sage-Bionetworks.github.io/governanceGreenPaper/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://Sage-Bionetworks.github.io/governanceGreenPaper/v/748aae7203ee269f6912798e0fc0cd8a1a7d94be/" />
  <meta name="manubot_html_url_versioned" content="https://Sage-Bionetworks.github.io/governanceGreenPaper/v/748aae7203ee269f6912798e0fc0cd8a1a7d94be/" />
  <meta name="manubot_pdf_url_versioned" content="https://Sage-Bionetworks.github.io/governanceGreenPaper/v/748aae7203ee269f6912798e0fc0cd8a1a7d94be/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Mechanisms to Govern Responsible Sharing of Open Data: A Progress Report</h1>
</header>
<nav id="TOC" role="doc-toc">
<ul>
<li><a href="#abstract">Abstract</a></li>
<li><a href="#introduction-to-data-governance">Introduction to Data Governance</a>
<ul>
<li><a href="#key-governance-concepts">Key Governance Concepts</a></li>
</ul></li>
<li><a href="#governance-structures">Governance Structures</a></li>
<li><a href="#governance-design-patterns">Governance Design Patterns</a>
<ul>
<li><a href="#types-of-governance-design-patterns">Types of Governance Design Patterns</a></li>
<li><a href="#pre-existing-governance-design-patterns">Pre-Existing Governance Design Patterns</a></li>
<li><a href="#transitioning-data-across-governance-structures">Transitioning Data Across Governance Structures</a></li>
<li><a href="#illustrative-use-cases">Illustrative use cases</a></li>
<li><a href="#governance-and-design-in-context">Governance and Design In Context</a></li>
</ul></li>
<li><a href="#conclusion">Conclusion</a>
<ul>
<li><a href="#epilogue">Epilogue</a></li>
</ul></li>
<li><a href="#acknowledgments">Acknowledgments</a></li>
<li><a href="#references">References</a></li>
</ul>
</nav>
<p><small><em>
This manuscript
(<a href="https://Sage-Bionetworks.github.io/governanceGreenPaper/v/748aae7203ee269f6912798e0fc0cd8a1a7d94be/">permalink</a>)
was automatically generated
from <a href="https://github.com/Sage-Bionetworks/governanceGreenPaper/tree/748aae7203ee269f6912798e0fc0cd8a1a7d94be">Sage-Bionetworks/governanceGreenPaper@748aae7</a>
on June 18, 2020.
</em></small></p>
<h2 class="unlisted" id="authors">Authors</h2>
<ul>
<li><p><strong>Lara Mangravite</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7841-3612">0000-0001-7841-3612</a><br>
<small>
Sage Bionetworks
</small></p></li>
<li><p><strong>Avery Sen</strong><br><br>
<small>
Sentripetal
</small></p></li>
<li><p><strong>John T. Wilbanks</strong><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-4510-0385">0000-0002-4510-0385</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/wilbanks">wilbanks</a><br>
<small>
Sage Bionetworks
</small></p></li>
</ul>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>This report provides a landscape analysis of of models of governance for open data sharing based on our observations in the biomedical sciences. We offer an overview of those observations and show areas where we think this work can expand to supply further support for open data sharing outside the sciences.</p>
<p>The central argument of this paper is that the “right” system of governance is determined by first understanding the nature of the collaborative activities intended. These activities map to types of <strong>governance structures</strong>, which in turn can be built out of standardized parts — what we call <strong>governance design patterns</strong>. In this way, governance for data science can be easy to build, follow key laws and ethics regimes, and enable innovative models of collaboration. We provide an initial survey of structures and design patterns, as well as examples of how we leverage this approach to rapidly build out ethics-centered governance in biomedical research.</p>
<p>As this paper itself will be deposited in GitHub, we also envision a contributory process whereby this inventory can be extended with more resources and links over time. We can envision communities using these design resources to create clearly governed networks. We can also imagine small private collectives amongst corporations and their partners, both academic and smaller businesses, using these designs as a “stack” to govern data science beyond biomedical research.</p>
<p>While there is no one-size-fits-all solution, we argue for learning from ongoing data science collaborations and building on from existing standards and tools. And in so doing, we argue for data governance as a discipline worthy of expertise, attention, standards, and innovation.</p>
<h2 class="page_break_before" id="introduction-to-data-governance">Introduction to Data Governance</h2>
<p>We have access to more data than ever before. We have access to faster computing resources than ever before. But technology, by itself, will not cures or create the solutions to climate change. The 2010s showed that unrestricted personal data surveillance can hack at the roots of society itself — changing elections and undermining collective trust in truth and experts.<span class="citation" data-cites="1GcAkA1FX"><sup><a href="#ref-1GcAkA1FX" role="doc-biblioref">57</a></sup></span> Data companies are now struggling to institute governance after-the-fact, building advisory boards, and attaching flags to posts.</p>
<p>Beyond social media and venture capital, however, lies a vast, rich, and often under-explored field of data governance. Particularly in the sciences, it is not controversial to believe that creating technology requires also creating the practices that govern its use. These scientific cultures often already work under regulation, and thus their existing practices predict life under data protection regulations and privacy laws.</p>
<p>Data scientists in the 21st century, like physicists in the last one, must reckon with the power they have to change the world forever. Data about people and the world is inseparable from the values that underlie its creation and application. But data scientists, particularly in non-regulated settings, often work without institutions, guideposts, or guardrails — without governance. The outcome is a mishmash of algorithmic inference that is inaccurate at best, racist and divisive at worst.<span class="citation" data-cites="M5iKzYGg"><sup><a href="#ref-M5iKzYGg" role="doc-biblioref">7</a></sup></span></p>
<p>Governance is a broad term. In a general sense, governance is a system of setting policy to encourage and prohibit behaviors, to monitor and enforce such policy, and to issue penalties or rewards accordingly. Governance is embodied in laws or other rules to tell humans (or machines) what they cannot do, or to incentivize them to try to do things they might not do otherwise. It can also be hardwired into tools (e.g. choice of data type) to prevent them from ever being able to do prohibited things, or to only be able to do things that are encouraged. Some aspects of governance are explicit — either codified into rules (written in natural language or computer code) or embodied in the very structure of equipment. Other aspects of governance are implicit — existing as cultural norms or tacit knowledge passed from person to person in practice.</p>
<p>In the context of biomedical research, we define governance as <strong>the freedoms, constraints, and incentives that determine how two or more parties manage the ingress, storage, analysis, and egress of data, tools, methods, and knowledge amongst themselves and with others</strong>.</p>
<p>Each step requires software, storage, compute power, know-how, and access to external digital resources. Each step further involves communication and negotiation. The resulting co-created knowledge is also more than just a publication in an academic journal, comprising mathematical models, networks, graphs, or other complex analyses, systems, or representations. Validating the claims in that knowledge may require access to analytic scripts written expressly for the data as well as enough infrastructure to re-run the entire analysis. The “data” in data science thus means more than just a literal data file.</p>
<p>Complicating matters further, different scientific communities manage governance very differently — high energy physics is the ultimate collaborative and standards-based field for governance, while in the field of biology governance remains individualized and often artisanal. This difference in governance intertwines with the nature of data and its collection: physicists are long accustomed to sharing large-scale equipment to generate, while biologists typically work in individualized laboratory settings with local data-generating equipment.<span class="citation" data-cites="11z0HOBna"><sup><a href="#ref-11z0HOBna" role="doc-biblioref">31</a></sup></span></p>
<p><strong>This paper argues that the “right” system of governance (including the “right” types and quantities of resources for governing) is determined by the nature of the collaborative activities intended</strong>. Concordantly, if our current system of governance is misaligned with our collaborative intent, and we want to transition to a system that is better aligned, then our transition strategy must be determined by an understanding of both the “as-is” and the “to-be” forms of collaboration.</p>
<p>As we will see in greater detail below, much scientific data governance revolves around how <em><strong>available</strong></em> data will be (i.e., how many and what types of people can access it) and how many <em><strong>freedoms</strong></em> are given to those who can access it (i.e., what conditions limit how can they use it). To understand these attributes of collaboration in context, consider the following examples.</p>
<p>First, a sensitive data set composed of a million mammograms with identifiable information could be powerful for studying breast cancer, if made widely available. But such a data set is enormous, costing tens of thousands of dollars in cloud fees just to transfer and store it, and more to analyze it. Further, its privacy implications would make its distribution complex, significantly constricting users. The right data governance approach can unlock that data set while balancing cost and privacy, allowing rigorously vetted users to apply for access to compute on it, privately, while preventing data extraction<span class="citation" data-cites="odIgeu98"><sup><a href="#ref-odIgeu98" role="doc-biblioref">51</a></sup></span> and undesired uses.<span class="citation" data-cites="nw5braAS"><sup><a href="#ref-nw5braAS" role="doc-biblioref">28</a></sup></span></p>
<p>Similarly, a data set that extracts sensor information about Parkinson’s disease from the phones of 15,000 people could drive new insights both into how the disease progresses, and into how we can use phones to study disease. We can study memory, phonation, and gait over time with no devices other than a smartphone. Such a data set would be far less identifiable than the mammograms, and smaller. Thus, its data governance can, in turn, leverage broader distribution and lower barriers to access, allowing a large audience of lightly vetted users to download, process, and publish new insights.<span class="citation" data-cites="LArYGREf"><sup><a href="#ref-LArYGREf" role="doc-biblioref">6</a></sup></span></p>
<p>These two examples are real-life examples from Sage Bionetworks (the DREAM digital mammography challenge and the mPower observational study of Parkinsons). They connect fundamentally through the conviction that there is not a simple answer to how we might govern data. Instead, we must look to technical realities, contractual methods, and, most importantly, how they can work together to unlock responsible data use in the service of more reliable claims from data science.</p>
<p>Importantly, our own focus is in biomedical research data sharing. This focus invokes a series of privacy laws and regulations relating to the identifiability of the data — its mismanagement has done direct harm to people<span class="citation" data-cites="ZvwpEFTA"><sup><a href="#ref-ZvwpEFTA" role="doc-biblioref">24</a></sup></span> and the variable nature of the data makes aggregation from across multiple sources complex. For us, the context under which data were collected will bound the ways the data can be meaningfully used. Thus, data sharing in our space regularly requires nuance and customization.<span class="citation" data-cites="10J8KhM0p"><sup><a href="#ref-10J8KhM0p" role="doc-biblioref">10</a></sup></span> While our space has been an outlier compared to consumer data and government open data, the increasing likelihood of data protection legislation around the globe means our examples may reflect a looming future for all data about individuals.</p>
<p>The good news is that these examples also point to a designable future that leverages a few key governance models to capture most of the common governance patterns encountered, with customization around the edges to reach the most complex patterns specific to any particular data set. We introduce the concept of <strong>governance structures</strong> to describe the models that we see forming over time. We also introduce the concept of <strong>governance design patterns</strong> to describe reusable elements of governance, such as standard contract language or user interfaces. These design patterns offer another path to governing open science: they codify a variety of research collaborations which, in turn, make more data more open. They represent a potential future direction for data governance as a discipline.</p>
<p>The rest of the paper will be as follows. We will account for major structures of scientific collaboration, including how suited each structure is to meeting open science goals. We will detail each governance structure in terms of a few key attributes, including availability and freedom. Next, we will discuss the governance design patterns associated with each of the collaborative patterns. This will lead into a discussion about how to transition data resources from one governance design to another — a form of data publication. Finally, we offer two example projects currently underway at Sage Bionetworks to make concrete the preceding concepts of governance structures and governance design patterns.</p>
<h3 id="key-governance-concepts">Key Governance Concepts</h3>
<p>Collaborations can embody a mixture of different forms of data movement and exportation, freedoms and restrictions on analysis, and openness and closure. In practice, we find that parties hold different ideas of what open, closed, accessible, findable, and interoperable actually mean. We are inspired by the FAIR principles — meaning an analysis of data as Findable, Accessible, Interoperable, Reusable — that are widely adopted in the life sciences,<span class="citation" data-cites="6DjakjNS"><sup><a href="#ref-6DjakjNS" role="doc-biblioref">63</a></sup></span> but their strength as a general standard also limits their application inside our own governance requirements. In this section, we will explain our own definitions of these and other terms.</p>
<p>We have chosen Availability and Freedoms as our two key attributes for characterizing collaborations, recognizing that both terms contain multitudes and, in some cases, may conflict with existing uses.</p>
<blockquote>
<p>Availability: the size of the population to whom a specific data set is available.</p>
</blockquote>
<p>Our first attribute is the <strong>total potential size of the user base</strong> for a given data set. This allows us to segment various kinds of collaboration projects by a clear metric of user population size. Here we examine concepts such as: exclusion of classes of users based on their employment (e.g., non-commercial restrictions) or training (e.g., allowance for formally trained and federally funded scientists), and requirements for transaction costs, such as registration, statements of intended use, and identity validation. Community scientists from non-traditional research settings are often excluded by these classes when concepts of availability are not intentionally designed into governance structures.</p>
<blockquote>
<p>Freedoms: the scale of the constraints under which a user must work.</p>
</blockquote>
<p>Thus, our second attribute concerns the ways that governance <strong>grants or restricts users’ rights to use the data</strong>. This measure is more qualitative than user population size, and forms more of a heuristic than a metric. Questions addressed include: Can the data be downloaded or redistributed? Are there fields of inquiry that are excluded? Is ethical oversight needed? Is exploratory use granted? And so forth.</p>
<blockquote>
<p>Open: the data are available under an explicit, pre-negotiated license that guarantees, at minimum, their ability to be downloaded and used for unrestricted data analysis.</p>
</blockquote>
<p>This definition focuses on granting, in advance, the rights to reuse data locally, and includes open licenses such as Creative Commons Zero, the Open Database License, and Microsoft’s Open Use of Data Agreement.</p>
<blockquote>
<p>Closed: the data are only available via petition to the data owner.</p>
</blockquote>
<p>This definition encompasses most day-to-day data practice.</p>
<blockquote>
<p>Restricted: there are regulatory or ethical restrictions on the data that necessarily constrain their availability.</p>
</blockquote>
<p>This definition contemplates, in particular, privacy and data protection law, but also constraints such as those present on data from indigenous peoples and tribal nations.</p>
<blockquote>
<p>Governance structure: a form of research collaboration governance, as characterized by the number and nature of relationships among parties, the relative degrees of availability of data and freedoms to use them, and instantiated by a contract (or other legal language) plus other, subordinate modules (e.g., user qualification mechanisms).</p>
</blockquote>
<p>This definition asserts that governance takes forms over time that are observable and repeatable.</p>
<blockquote>
<p>Governance Design Pattern (or simply “pattern”): a module within a governance structure, i.e., a concrete artifact (e.g., boilerplate language for rules, or algorithms to data monitor sharing).</p>
</blockquote>
<p>This definition asserts that governance structures are composed of specific and reusable patterns.</p>
<h2 id="governance-structures">Governance Structures</h2>
<p>Effective data use traditionally starts not with, “what data do we have?” but instead with, “what hypothesis do we propose to explore?” The advent of machine learning adds to this, “what hypotheses are afforded by these data?” These two questions create specific local cases with wide and diverse requirements: what data to acquire, how to bring them into systems, how to store them, how to analyze them, how to share downstream knowledge.</p>
<p>Because of this diversity, what is appropriate data handling in one project may not be appropriate in another. Notably, open approaches to data access do not consistently lead to their responsible and reliable reuse.<span class="citation" data-cites="6j990tze"><sup><a href="#ref-6j990tze" role="doc-biblioref">46</a></sup></span> This has made the process of data governance itself difficult to “open source” — simply giving away a clinical protocol, data management process, or metadata schema under an open source license does not ensure meaningful data reuse.</p>
<p>Openness itself merits interrogation. A key requirement for data governance to consider in this context is: <strong>what does “open” mean to different people, and different groups, over time</strong>? Openness does not work for everyone, everywhere. Many groups traditionally under-represented in medical research have endured systemic, ongoing betrayals of trust in how data and samples are collected and used.<span class="citation" data-cites="H3YsDbZv"><sup><a href="#ref-H3YsDbZv" role="doc-biblioref">5</a></sup></span> These same groups are also often the least served by the medical systems, and the most surveilled by the state.<span class="citation" data-cites="EIBAJZJB"><sup><a href="#ref-EIBAJZJB" role="doc-biblioref">20</a></sup></span> We must therefore design governance structures and governance design patterns that support the contextual desire to restrict data availability.<span class="citation" data-cites="vzr6hitM"><sup><a href="#ref-vzr6hitM" role="doc-biblioref">49</a></sup></span> We must also develop the ability bring every group into governance designs, to explore when, how, and to whom they might be comfortable allowing access over time, and governance appropriate for those uses.<span class="citation" data-cites="Piezz6R5"><sup><a href="#ref-Piezz6R5" role="doc-biblioref">62</a></sup></span></p>
<p>While research collaborations are unique, there are commonalities they share. At Sage Bionetworks, we have observed regular, relatively stable structures that work well over time. We will refer to these as <strong>governance structures</strong>.</p>
<p>The major governance structures we have observed are the following:</p>
<ul>
<li><strong>Pairwise</strong> (One-to-one): Two parties agree to work together and/or share on a data set in some fashion, typically with a closed contract or an informal agreement. The negotiation terms depend on the relative status of the parties and/or the value of the data and knowledge.</li>
<li><strong>Open source</strong> (One-to-many or some-to-many): data are distributed for reuse with a license defining reuse rights and conditions. The creator is in charge of the negotiation at first (choice of license), but then rights to analyze and redistribute are permanently transferred to the user. This is typical of a centralized project in the sciences, i.e., the Human Genome Project.<span class="citation" data-cites="32SWh1ey"><sup><a href="#ref-32SWh1ey" role="doc-biblioref">11</a></sup></span></li>
<li><strong>Federated Query</strong> (Many-to-many, via platform): Data are housed in a variety of locations, and users are able to query to those local data simultaneously. Typically restricted to pre-configured queries (rather than data exploration) and may require registration before use.<span class="citation" data-cites="TQlIEC6o"><sup><a href="#ref-TQlIEC6o" role="doc-biblioref">52</a></sup></span></li>
<li><strong>Trusted research environment</strong> (Many-to-some): data are housed in a central location under a contractual regime including data transfer and use agreements. Users apply to use the data. Users must “visit” the data rather than download them, agree to be known, and, in some cases, agree to be surveilled by a data steward.<span class="citation" data-cites="MmeIANuu"><sup><a href="#ref-MmeIANuu" role="doc-biblioref">32</a></sup></span></li>
<li><strong>Model-to-data</strong> (One-to-many): Data are held by a steward who is responsible for running algorithms on the behalf of researchers. In some cases, a synthetic version of the data may be released openly to facilitate model training. Researchers develop algorithms, send them to the steward, and receive back output of their analysis as run on the real dataset. The variety of analyses that may be performed is restricted by this structure, because the data steward must ensure data are specifically curated for any analytical question at hand.<span class="citation" data-cites="nw5braAS"><sup><a href="#ref-nw5braAS" role="doc-biblioref">28</a></sup></span></li>
<li><strong>Open citizen science</strong> (Many-to-many): Rights to use and distribute data are often fully decentralized via license or contract. Open citizen science is a peer-to-peer version of open source science.<span class="citation" data-cites="12HajymAr"><sup><a href="#ref-12HajymAr" role="doc-biblioref">26</a></sup></span></li>
<li><strong>Clubs and Trusts</strong> (Some-to-some): This is a common version of collaboration in biomedical research. Clubs<span class="citation" data-cites="abUbT01W"><sup><a href="#ref-abUbT01W" role="doc-biblioref">45</a></sup></span> and Trusts<span class="citation" data-cites="TUd0l6ty"><sup><a href="#ref-TUd0l6ty" role="doc-biblioref">36</a></sup></span> are versions of a common pool resource: a group of people and/or institutions who agree to share resources towards a common goal. Control over the development and negotiation of data sharing and use terms is often held by the founders / settlers (and/or funders) and then can be distributed amongst club participants.<span class="citation" data-cites="abUbT01W"><sup><a href="#ref-abUbT01W" role="doc-biblioref">45</a></sup></span> Importantly, <strong>clubs that operate in the cloud can easily publish data products that are more “open” than the club itself</strong>.</li>
<li><strong>Closed</strong>: Data are held privately by a single party.</li>
<li><strong>Closed and Restricted</strong>: Data are held privately in order to protect a population, meet a legal requirement, or protect a secret.</li>
</ul>
<div id="tbl:structures" class="tablenos">
<table style="width:100%;">
<caption><span>Table 1:</span> Governance structures and their attributes
</caption>
<colgroup>
<col style="width: 16%" />
<col style="width: 16%" />
<col style="width: 16%" />
<col style="width: 16%" />
<col style="width: 16%" />
<col style="width: 16%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Governance structure</strong></th>
<th>Number and linkage of parties</th>
<th>Degree of data Availability</th>
<th>Degree of freedom to use data</th>
<th>Challenges common to the governance success</th>
<th>Primary governance design pattern</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>Pairwise</strong></td>
<td>One-to-one</td>
<td>Medium/High</td>
<td>Medium/High</td>
<td>Uneven status of parties, value of data</td>
<td>Informal or closed contract</td>
</tr>
<tr class="even">
<td><strong>Open Source</strong></td>
<td>One/some-to-many</td>
<td>High</td>
<td>High</td>
<td>Rights permanently granted to user</td>
<td>License</td>
</tr>
<tr class="odd">
<td><strong>Federated Query</strong></td>
<td>Many-to-many, via platform</td>
<td>High</td>
<td>Medium/Low</td>
<td>Defection of creators</td>
<td>Contract and club rules</td>
</tr>
<tr class="even">
<td><strong>Trusted Research Environment</strong></td>
<td>One/some-to-many</td>
<td>Medium/Low</td>
<td>Medium/Low</td>
<td>Users agree to be known, surveilled</td>
<td>Data transfer and use agreements</td>
</tr>
<tr class="odd">
<td><strong>Model-to-Data</strong></td>
<td>One-to-many</td>
<td>High</td>
<td>Low</td>
<td>Not all who apply can use data</td>
<td>Restricted analyses, data curation</td>
</tr>
<tr class="even">
<td><strong>Open Citizen Science</strong></td>
<td>Many-to-many</td>
<td>High</td>
<td>High</td>
<td>Capacity for analysis is uneven</td>
<td>Contract or license</td>
</tr>
<tr class="odd">
<td><strong>Clubs, Trusts</strong></td>
<td>Some-to-some</td>
<td>Medium/Low</td>
<td>High</td>
<td>Easy to create things governed more liberally. Trusteeship can be revoked.</td>
<td>Club / Trust rules</td>
</tr>
<tr class="even">
<td><strong>Closed</strong></td>
<td>Many (to none)</td>
<td>Low</td>
<td>High</td>
<td>Fundamental limits to collaboration</td>
<td>Public laws, security protocols</td>
</tr>
<tr class="odd">
<td><strong>Closed and Restricted</strong></td>
<td>Some (to none)</td>
<td>Low</td>
<td>Low</td>
<td>Fundamental limits to collaboration</td>
<td>Public laws, security protocols</td>
</tr>
</tbody>
</table>
</div>
<p>How many parties are involved, who is in charge of the negotiation, how significantly are the data regulated or the sensitive nature of the data — these are all attributes that can be used to characterize the governance structure. From the many potential attributes, we choose to organize governance structures by a) how broadly available they are — the total potential size of the user base — and b) how many freedoms a user has to freely use and distribute the data.</p>
<p>These two attributes are “upstream” from decisions such as which license to use, and form a pair of axes on which to orient structural governance analysis. These attributes provide a standardized form to describe governance structures in which one, some, or many parties either provide or use data, and the freedom to use and distribute data is variable.</p>
<div id="fig:fig1" class="fignos">
<figure>
<img src="images/figure1.png" style="width:80.0%" alt="" /><figcaption><span>Figure 1:</span> Governance structures and their relationship types, by relative amounts of availability and freedom</figcaption>
</figure>
</div>
<hr />
<div id="fig:fig2" class="fignos">
<figure>
<img src="images/figure2.png" style="width:80.0%" alt="" /><figcaption><span>Figure 2:</span> Governance structure and process flow for pairwise collaborations</figcaption>
</figure>
</div>
<hr />
<div id="fig:fig3" class="fignos">
<figure>
<img src="images/figure3.png" style="width:80.0%" alt="" /><figcaption><span>Figure 3:</span> Governance structure and process flow for open source collaborations</figcaption>
</figure>
</div>
<hr />
<div id="fig:fig4" class="fignos">
<figure>
<img src="images/figure4.png" style="width:80.0%" alt="" /><figcaption><span>Figure 4:</span> Governance structure and process flow for federated query</figcaption>
</figure>
</div>
<hr />
<div id="fig:fig5" class="fignos">
<figure>
<img src="images/figure5.png" style="width:80.0%" alt="" /><figcaption><span>Figure 5:</span> Governance structure and process flow for model-to-data</figcaption>
</figure>
</div>
<hr />
<div id="fig:fig6" class="fignos">
<figure>
<img src="images/figure6.png" style="width:80.0%" alt="" /><figcaption><span>Figure 6:</span> Governance structure and process flow for trusted researcer environment</figcaption>
</figure>
</div>
<p>Governance structures are abstractions, not turnkey solutions that we can provide to would-be collaborators, whether open or closed. These structures are most useful to orient data collaborations: do we want a club, or a trusted environment? By asking these questions about how available data will be, and how much freedom a user will have to analyze and distribute, we can quickly identify the goals of the collaborating parties and move into drafting contract structures.</p>
<p>These structures employ governance design patterns: standard contracts and contractual language, user interfaces, teaching toolkits, and software templates. <strong>These patterns can be assembled into a governance structure faster than drafting from scratch, while leveraging legal rigor of past work for regulated or protected data collaboration</strong>.</p>
<div id="fig:fig7" class="fignos">
<figure>
<img src="images/figure7.png" style="width:80.0%" alt="" /><figcaption><span>Figure 7:</span> Decisions about governance should start with research goals and nature of collaboration, then move to what structure and design elements are needed.</figcaption>
</figure>
</div>
<h2 id="governance-design-patterns">Governance Design Patterns</h2>
<p>Like templates for building a spreadsheet or website, <strong>governance design patterns are generalizable solutions for common challenges and decision points in collaboration</strong>.</p>
<p>Governance design patterns are often not finished products that can be transformed directly into contract. Any real-life governance structure will combine nuances about specific data transfer and use with existing design patterns such as standards, toolkits, reference implementations, process knowledge, and pedagogy. Governance design patterns themselves must often be customized, or “filled out,” from their default forms. Authors of these design patterns can build maps of what kinds of governance design patterns work well with each other, which don’t, and which, in turn, serve as inputs into new design patterns.</p>
<h3 id="types-of-governance-design-patterns">Types of Governance Design Patterns</h3>
<p>Data collaborations in biomedical research often feature scientists negotiating legal language directly. Their selection, combination, and application of legal terms is often ad-hoc, often leading to missing language, conflicting language, or misinterpretation. Standardizing contract language patterns and defining their interactions creates a common language for governance early in collaborations.</p>
<p>More importantly, standard, yet customizable, reusable language creates multiple benefits. Such language is not biased to anyone in the negotiation, meets legal and ethical norms and requirements often missed by unskilled negotiators, and supports descriptive annotations to facilitate understanding. Standard language also supports automated data governance regimes such as machine review of data use agreements.<span class="citation" data-cites="1DGkp34ZB"><sup><a href="#ref-1DGkp34ZB" role="doc-biblioref">18</a></sup></span></p>
<p>Therefore, <strong>the primary governance design pattern at Sage Bionetworks is legal language, usually (but not exclusively) situated inside a contract or a license</strong>. Other patterns include teaching toolkits and user interfaces.</p>
<p>There is a wide range of promising areas to explore for establishing standard governance designs and their constituent design patterns. We review these areas here.</p>
<h4 id="licenses-and-agreements">Licenses and Agreements</h4>
<p><strong>Licenses, contractual agreements, and regulatory documents</strong> can be standard documents, standard paragraphs to be inserted into custom documents, or templates with significant content filled in by the users.</p>
<div id="tbl:licensesAgreements" class="tablenos">
<table>
<caption><span>Table 2:</span> Licenses and agreements
</caption>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th>Design Pattern</th>
<th>Description</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Open license</td>
<td>A license or contract which contains provisions that allow other individuals to reuse data with specific freedoms and conditions. Connected to intellectual property laws.</td>
<td>Creative Commons,<span class="citation" data-cites="116CDX0ks"><sup><a href="#ref-116CDX0ks" role="doc-biblioref">54</a></sup></span> Open Database License,<span class="citation" data-cites="1A5Jsy5oq"><sup><a href="#ref-1A5Jsy5oq" role="doc-biblioref">42</a></sup></span> Microsoft Open Use of Data Agreement<span class="citation" data-cites="zCTNJz8u"><sup><a href="#ref-zCTNJz8u" role="doc-biblioref">38</a></sup></span></td>
</tr>
<tr class="even">
<td>Data use agreement</td>
<td>An agreement that allows use of restricted data in a specific way. Connected to health privacy laws.</td>
<td>Health Care Systems Network Data Use Agreement Toolkit<span class="citation" data-cites="j7nvT8JA"><sup><a href="#ref-j7nvT8JA" role="doc-biblioref">29</a></sup></span></td>
</tr>
<tr class="odd">
<td>Data transfer agreement</td>
<td>An agreement that allows transfer of data from one party to another not associated with the first party under certain terms and conditions. Connected to data protection laws.</td>
<td>European Commission Standard Contractual Clauses,<span class="citation" data-cites="wthFx3D2"><sup><a href="#ref-wthFx3D2" role="doc-biblioref">55</a></sup></span> Uniform Biological Materials Transfer Agreement<span class="citation" data-cites="t5HXYtyn"><sup><a href="#ref-t5HXYtyn" role="doc-biblioref">34</a></sup></span></td>
</tr>
<tr class="even">
<td>Data processing addendum</td>
<td>Standard contractual clauses for transfer between data controllers and data processors. Relevant in GDPR transfers as well as emerging state legislation in the US.<span class="citation" data-cites="12mne7E5t"><sup><a href="#ref-12mne7E5t" role="doc-biblioref">1</a></sup></span></td>
<td>IAPP Sample Addendum Addressing Article 28 GDPR and Incorporating Standard Contractual Clauses for Controller to Processor Transfers of Personal Data<span class="citation" data-cites="e167exJ6"><sup><a href="#ref-e167exJ6" role="doc-biblioref">50</a></sup></span></td>
</tr>
<tr class="odd">
<td>Public domain</td>
<td>A declaration that no intellectual property rights exist in a data set or database.</td>
<td>Creative Commons CC0,<span class="citation" data-cites="bQ5U0G1W"><sup><a href="#ref-bQ5U0G1W" role="doc-biblioref">13</a></sup></span> Public Domain Database License<span class="citation" data-cites="f8vwfMHl"><sup><a href="#ref-f8vwfMHl" role="doc-biblioref">43</a></sup></span></td>
</tr>
</tbody>
</table>
</div>
<h4 id="data-collection-and-ingest-mechanisms">Data Collection and Ingest Mechanisms</h4>
<p>Data has to be ingested into systems, so that it can be connected to other data, analytic framework, backup, and encryption. The <strong>data ingest design patterns typically take the form of standard paragraphs that are used in contracts, filings (such as clinical protocols), and other legal documents</strong>.</p>
<p>For example, in a memorandum of understanding, each side might specify which technical standards will be used in which parts of the project, or one side might ask the other to assert a level of cybersecurity that meets federal standards.</p>
<p>Depending on the data type, regulations may require informed consent, simple consent, and compliance with state and national data protection laws. These legal documents execute a basic set of functions: gathering permission from a person to take in their data into a research study, cloud platform, business, and applications. Broadly standard language dominates consumer terms of service and privacy policy, with outright cut-and-paste common from website to website. A primary goal of most of these documents is to insulate the collector from liability and the resulting documents are complex, lawyer-oriented, difficult to understand, and non-negotiable. Thus the need for user interfaces to explain data collection is increasingly accepted, with user experience designs demonstrating impact in informed consent and privacy policies.</p>
<div id="tbl:collectionMechanisms" class="tablenos">
<table>
<caption><span>Table 3:</span> Data collection mechanisms
</caption>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th>Design Pattern</th>
<th>Description</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Clinical protocol</td>
<td>A written plan for how a health condition, drug, or device is to be studied and the procedure to be followed by the study, including technology. Connected to international and national laws in bioethics.</td>
<td>CDISC Clinical Trial Protocol Representation Model,<span class="citation" data-cites="hNKKknXk"><sup><a href="#ref-hNKKknXk" role="doc-biblioref">47</a></sup></span> NIH Protocol Templates for Clinical Trials<span class="citation" data-cites="CxfmIM8F"><sup><a href="#ref-CxfmIM8F" role="doc-biblioref">48</a></sup></span></td>
</tr>
<tr class="even">
<td>Informed consent</td>
<td>Standard language and user interfaces that have been vetted through legal and ethical review for specific types of data, e.g. health records or DNA</td>
<td>Sage Bionetworks Elements of Informed Consent,<span class="citation" data-cites="11WcMMgG1"><sup><a href="#ref-11WcMMgG1" role="doc-biblioref">19</a></sup></span> GA4GH Model Consent Clauses<span class="citation" data-cites="1DGkp34ZB"><sup><a href="#ref-1DGkp34ZB" role="doc-biblioref">18</a></sup></span></td>
</tr>
<tr class="odd">
<td>Privacy policy</td>
<td>Standard language and user interfaces that have been vetted through legal and ethical review for apps and websites that collect user data</td>
<td>Sage Bionetworks Privacy Toolkit<span class="citation" data-cites="1Cx91jrTR"><sup><a href="#ref-1Cx91jrTR" role="doc-biblioref">60</a></sup></span></td>
</tr>
<tr class="even">
<td>Terms of service</td>
<td>Standard language and user interfaces that have been vetted through legal review for apps and websites that collect user data</td>
<td>Various auto-contract systems e.g. Formswift,<span class="citation" data-cites="ENjTbvYJ"><sup><a href="#ref-ENjTbvYJ" role="doc-biblioref">22</a></sup></span> LegalZoom<span class="citation" data-cites="G3NU3hO7"><sup><a href="#ref-G3NU3hO7" role="doc-biblioref">33</a></sup></span></td>
</tr>
</tbody>
</table>
</div>
<h4 id="user-qualification-mechanisms">User qualification mechanisms</h4>
<p><strong>Data analysis governance design patterns center around how a given project admits users</strong>. While closed projects don’t allow user admission at all (an anti-pattern), a broad range of scientific and consumer data holders use access committee review to decide when and if to allow a new usage of data. Committees traditionally perform both positive review (“does this researcher meet the minimum requirements?”) and normative review (“is this use worthy?”).<span class="citation" data-cites="VG5MXB5B"><sup><a href="#ref-VG5MXB5B" role="doc-biblioref">9</a></sup></span> More recently, major biomedical collaborations began testing how broad usage rights granted objectively to users might work, especially in cloud computing.<span class="citation" data-cites="80a55iFY"><sup><a href="#ref-80a55iFY" role="doc-biblioref">15</a></sup></span></p>
<div id="tbl:userQual" class="tablenos">
<table>
<caption><span>Table 4:</span> User qualification mechanisms
</caption>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th>Design Pattern</th>
<th>Description</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Data Access Committee (DAC)</td>
<td>A committee, whether a formal or informal group of individuals, with the responsibility of reviewing and assessing data access requests. Many individual groups, consortiums, institutional and independent DACs have been established but there is currently no widely accepted framework for their organization and function</td>
<td>dbGAP<span class="citation" data-cites="BtzWfCJm"><sup><a href="#ref-BtzWfCJm" role="doc-biblioref">39</a></sup></span></td>
</tr>
<tr class="even">
<td>Ethics and Regulatory Review Boards</td>
<td>A formal accredited committee with the responsibility of reviewing and assessing research involving human beings.<span class="citation" data-cites="1DMYQt6hx"><sup><a href="#ref-1DMYQt6hx" role="doc-biblioref">35</a></sup></span> Research institutions and governments typically run their own review boards, while for profit and other independent review boards provide the same services for a fee to the public. These boards review clinical protocols, informed consent forms, and other study related material.</td>
<td>AllofUs Research Program Institutional Review Board,<span class="citation" data-cites="1HXhjeVQY"><sup><a href="#ref-1HXhjeVQY" role="doc-biblioref">3</a></sup></span> WCG-WIRB<span class="citation" data-cites="1EZbXJTg3"><sup><a href="#ref-1EZbXJTg3" role="doc-biblioref">64</a></sup></span></td>
</tr>
<tr class="odd">
<td>Qualified Researcher</td>
<td>A process by which researchers can apply to use data with a range of conditions and freedoms. Reconstructs some functionality of open licenses while allowing data download.<span class="citation" data-cites="1As3pvk5w"><sup><a href="#ref-1As3pvk5w" role="doc-biblioref">25</a></sup></span></td>
<td>mPower Qualified Researcher Release<span class="citation" data-cites="otY3SefD"><sup><a href="#ref-otY3SefD" role="doc-biblioref">61</a></sup></span></td>
</tr>
<tr class="even">
<td>Data Passport</td>
<td>A process by which researchers can apply for general permission to do exploratory analysis of data with wide freedoms. Reconstructs some functionality of open licenses, which typically prohibits data download<span class="citation" data-cites="10nEXzoVA"><sup><a href="#ref-10nEXzoVA" role="doc-biblioref">2</a></sup></span> .</td>
<td>AllofUs Research Program Data Access Framework<span class="citation" data-cites="80a55iFY"><sup><a href="#ref-80a55iFY" role="doc-biblioref">15</a></sup></span></td>
</tr>
<tr class="odd">
<td>Registration</td>
<td>Data are available to users who complete a registration process and agree to terms of service. Can be used to construct a non commercial offering.</td>
<td>Catalog of Somatic Mutations in Cancer (COSMIC)<span class="citation" data-cites="1mFzspuo"><sup><a href="#ref-1mFzspuo" role="doc-biblioref">59</a></sup></span></td>
</tr>
<tr class="even">
<td>Open Download</td>
<td>Data are available on the public internet with no restrictions beyond those required to gain access to the internet and norms of attribution.</td>
<td>SyntheticMass FHIR Dataset,<span class="citation" data-cites="H7PziDQ7"><sup><a href="#ref-H7PziDQ7" role="doc-biblioref">58</a></sup></span> Center for Medicare Services dataset downloads<span class="citation" data-cites="xQxy7NV0"><sup><a href="#ref-xQxy7NV0" role="doc-biblioref">14</a></sup></span></td>
</tr>
</tbody>
</table>
</div>
<h4 id="mathematical-governance-mechanisms">Mathematical governance mechanisms</h4>
<p>A variety of technical approaches can constitute governance design patterns. De-identification is perhaps the oldest mathematical design pattern, and is incorporated into laws such as HIPAA as a key pattern of data governance. <strong>New mathematical approaches hold promise to enable extremely broad availability and freedom at low risk</strong>, but can be computationally expensive on enormous data sets in comparison to other forms of governance.</p>
<div id="tbl:mathGov" class="tablenos">
<table>
<caption><span>Table 5:</span> Mathematical governance mechanisms
</caption>
<colgroup>
<col style="width: 33%" />
<col style="width: 33%" />
<col style="width: 33%" />
</colgroup>
<thead>
<tr class="header">
<th>Design Pattern</th>
<th>Description</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>De-identification</td>
<td>Language in which two or more parties agree to use either removal of fields or mathematical techniques to achieve a standard of de-identification. Typically inserted into larger governance contracts.</td>
<td>De-identification sample clauses<span class="citation" data-cites="14K4iV022"><sup><a href="#ref-14K4iV022" role="doc-biblioref">17</a></sup></span> , HHS Sample Business Associate Agreements<span class="citation" data-cites="1BpKAGz3u"><sup><a href="#ref-1BpKAGz3u" role="doc-biblioref">41</a></sup></span></td>
</tr>
<tr class="even">
<td>Blockchain</td>
<td>Language by which two or more parties agree to use an open, distributed, permanent ledger to broker transactions and data use.</td>
<td>Accord project template studio<span class="citation" data-cites="H5KgzA16"><sup><a href="#ref-H5KgzA16" role="doc-biblioref">56</a></sup></span></td>
</tr>
<tr class="odd">
<td>Homomorphic encryption</td>
<td>Language in which two or more parties agree to use encryption that allows computation on distributed storage and computation while preserving privacy. Most use so far in emerging smart contracts regimes.<span class="citation" data-cites="5plIyD5u"><sup><a href="#ref-5plIyD5u" role="doc-biblioref">40</a></sup></span></td>
<td>N/A</td>
</tr>
<tr class="even">
<td>Differential privacy</td>
<td>Language by which two or more parties agree to publicly expose information about a dataset by describing patterns in the dataset without disclosing data about individuals. Most implementations cluster in technical projects,<span class="citation" data-cites="1F7XRqf7Z"><sup><a href="#ref-1F7XRqf7Z" role="doc-biblioref">44</a></sup></span> with little standard contract language available.</td>
<td>N/A</td>
</tr>
</tbody>
</table>
</div>
<h4 id="data-type">Data Type</h4>
<p>Data type — the kind of data in use — defines much of the universe of a collaboration structure. Regulation, legislation, civil rights, norms, and contracts are tied to specific kinds of data, such as electronic health records and DNA. Data types are most often negotiated among collaborators, and are easiest to see inside technical standards, file formats, transactions, and more. They are not themselves governance design patterns, although design patterns will make many references to them.</p>
<p>In our work at Sage Bionetworks, we most often deal with data about DNA and related biological processes, “real world” data we collect about people’s health through their phones and wearables, and electronic health records. There is also a vast universe of consumer data available under private contract, such as credit card or grocery data.</p>
<h3 id="pre-existing-governance-design-patterns">Pre-Existing Governance Design Patterns</h3>
<p>We know that standard contract language can have an impact in areas outside of data governance. The Federal Demonstration Partnership (FDP), a 35-year-old cooperative initiative among 10 federal agencies and 154 institutional recipients of federal funds, implemented a vast set of standard governance documents to help groups form, apply for, and share federal funding, and report back on the collaboration outcomes.<span class="citation" data-cites="AnAX84nk"><sup><a href="#ref-AnAX84nk" role="doc-biblioref">30</a></sup></span> An early FDP survey found that of the time that faculty committed to federal research, 42 percent was devoted to administrative activities rather than research. Researchers indicated that “grant progress-report submissions, personnel hiring, project-revenue management, equipment and supply purchases, IRB protocols and training, training personnel and students, and personnel evaluations” were the top drains on research time.<span class="citation" data-cites="wqupV8k1"><sup><a href="#ref-wqupV8k1" role="doc-biblioref">16</a></sup></span></p>
<p>The FDP years ago created what we — in this paper — call governance design patterns. Negotiations using the FDP contracts can be orders of magnitude simpler and faster because <strong>the standard design patterns can create trust and equality among negotiators</strong>, rather than lawyer-to-lawyer negotiation. However, a 2018 survey found a slight increase in time spent on administrative activity despite the FDP’s existence.<span class="citation" data-cites="isF1I1NM"><sup><a href="#ref-isF1I1NM" role="doc-biblioref">23</a></sup></span> This increase may be connected to the lack of support for researchers at institutions implementing negotiations.<span class="citation" data-cites="tPpeCQYm"><sup><a href="#ref-tPpeCQYm" role="doc-biblioref">37</a></sup></span> Thus design patterns themselves will not solve the problem: <strong>governance needs implementation, hiring, funding, and support to succeed</strong>.</p>
<p>The FDP is joined by the SMART IRB platform funded by NIH, which eases challenges associated with initiating multisite research and provides a roadmap for institutions to implement the NIH Single IRB Review policy. Clinical protocols and consent forms often reuse language from previous projects, replicating the problem of defending past approaches whether or not they meet the current issue. This creates similar value to standardized neutral language intended for customization, i.e., design patterns. Like the FDP, SMART IRB demonstrates the value of legal tools at the design pattern level in contracts and negotiation. This finding is replicated in the design of informed consent for national research studies.</p>
<p>We know from the FDP and SMART IRB examples that the standardization of language and process can ease the burden of drafting contracts for co-writing grants or for regulatory paperwork. But <strong>there is no similar unifying effort for governance design patterns (including, but not limited to legal language) in data-sharing governance</strong>. Nor is there sufficient recognition of the complexity of implementing these tools in practice. As the 2018 FDP survey found, simplifying the language does not automatically ease the administrative burden.</p>
<h3 id="transitioning-data-across-governance-structures">Transitioning Data Across Governance Structures</h3>
<p>Because we believe that open science makes for better science, <strong>we work to promote governance structures and design patterns that support data sharing in a manner that is as open as possible</strong>. But as we note above, not all data can or should be shared openly at all times.</p>
<p>We see value in exploring how data generated within one governance structure can migrate from a more closed context to a more open one. This allows us to define a more closed data use context for a small group of users, and export information and data to a more open structure as a form of publication. This balance is often impossible to achieve in a single governance structure: patterns that lead to reuse and patterns that lead to protection create internal conflict. The solution is to develop a program that uses a different design pattern for management of digital assets developed across the project lifecycle and/or to design a process that migrates certain data from one governance structure to the other under the right conditions. <em><strong>These transitions can create balance between very real needs to protect data during analysis and the very real opportunities that can come from increased access to data</strong></em>.</p>
<p>Our first example provided a different governance structure for sharing of data vs. sharing of analytical outputs. In this case, a private, pre-competitive group of cancer researchers working together on genetic subtypes for colorectal cancer perceived that sharing data with one another was worth the risk, but publishing the data — or the algorithms they build with that data — onto the public internet, under an open license, was not.<span class="citation" data-cites="KLKqRnt"><sup><a href="#ref-KLKqRnt" role="doc-biblioref">27</a></sup></span> This project used a club structure to enable data sharing so that each researcher could develop their algorithm with the extended set of data but, in exchange, the group retained the right to compare those algorithms and develop a consensus model that could be published openly. This shift in design patterns supports a transition from club to open structure for key digital assets in a manner that balances the public benefit of the collective wisdom with the private incentives of the individual researchers.</p>
<div id="fig:fig8" class="fignos">
<figure>
<img src="images/figure8.png" style="width:80.0%" alt="" /><figcaption><span>Figure 8:</span> Transitioning from one governance structure to another requires transitioning from one governance design pattern to another.</figcaption>
</figure>
</div>
<p>A second example illustrates a shift from a trusted environment to an open structure for data sharing over time. The Accelerating Medicines Partnership for Alzheimer’s Disease, a public private partnership between the US NIH and the pharmaceutical industry, engaged academic and pharmaceutical industry researchers in a pre-competitive program to identify new drug targets in Alzheimer’s from multi-omic brain data. The researchers agreed to share data, algorithms and outcomes with each other immediately upon data generation — and to periodically migrate those assets into the public domain upon a pre-specified schedule (ADknowledgeportal.org).</p>
<p>In a third example, the Breast Cancer Surveillance Consortium and the Icahn School of Medicine at Mount Sinai had a database of 640,000 mammograms. Wondering if machine learning might provide assistance to, or even match, expert radiologists, they hoped to convene a computational contest. But the data had not been consented for either high freedom or high availability, and bore risks of re-identification of the patients represented by the data. Beyond that, the scale of the data made broad access to it computationally and economically expensive. This required an analysis structure that was tightly controlled, but some form of data distribution that was liberal enough to support hundreds of data users.<span class="citation" data-cites="dIYXacYW"><sup><a href="#ref-dIYXacYW" role="doc-biblioref">53</a></sup></span></p>
<p>By creating a synthetic form of the data set — which could be downloaded widely, as it does not represent real people — a wide set of analysts could use this data to inform algorithm development. Acting as a steward, Sage Bionetworks could then run models on behalf of the submitters and return results and feedback to allow for model tuning, iteration, and development. This transition of synthetic data from the model-to-data structure to open data kept an “air gap” around the real data to protect its privacy, but allowed hundreds of users worldwide to build algorithms.<span class="citation" data-cites="nw5braAS"><sup><a href="#ref-nw5braAS" role="doc-biblioref">28</a></sup></span></p>
<h3 id="illustrative-use-cases">Illustrative use cases</h3>
<p>The COVID19 pandemic has forced an unprecedented spike in scientific data collaboration.</p>
<p>Our approach of using governance design patterns to rapidly build desired governance structures has, so far, performed well under time pressure. We illustrate here two use cases, the National COVID Cohort Collaborative and the COVID Recovery Corps, to demonstrate how we could stand up two different projects in less than two months without sacrificing legal quality or ethical rigor.</p>
<h4 id="national-covid-cohort-collaborative-n3c">National COVID Cohort Collaborative (N3C)</h4>
<p>The National COVID Cohort Collaborative (N3C) is a collaboration among 60 grant-receiving clinical centers and their partners, 5 pre-existing distributed health data networks, the NIH’s National Center for Advancing Translational Sciences (NCATS), and the pre-existing National Center for Data To Health project (CD2H). The N3C’s goal is to enable detailed data analysis of clinical records for COVID patients for a wide variety of users while preserving individual privacy of patients.</p>
<p>Formed rapidly in response to the outbreak of the pandemic, the N3C is assembling relevant electronic health record data of COVID-19 patients from participating institutions. The data live in a secure analytical environment where tools and algorithms can be rapidly evaluated, and clinicians and researchers can ask granular and complex clinical questions. Additionally, the same data will drive a synthetic data set available for broad download and reuse, and partners will expose their data for pre-written federated queries.</p>
<p>As members of the CD2H project, Sage Bionetworks governance leaders chose a suite of governance structures and transition: first, a club managed by NCATS into which data flows from the many data owners. Second, the N3C creates a range of transitions for data. Some users sign a data use agreement to get into the “real” data set, others to run federated queries. Synthetic data will be available for registered download. The entire process is run through a single ethical review board at Johns Hopkins University under SMART IRB.</p>
<p>By using governance structures and design patterns, the N3C was able to move governance rapidly while retaining high quality. Within two months of launch, the N3C protocol passed review at Johns Hopkins, more than 37 institutions have signed the single data transfer agreement, and data access is already in testing. Our governance aims to enable the project to achieve its goals to improve the efficiency and accessibility of analyses with COVID-19 clinical data, expand our ability to analyze and understand COVID, and demonstrate a novel approach for collaborative pandemic data sharing.</p>
<div id="fig:fig9" class="fignos">
<figure>
<img src="images/figure9.png" style="width:80.0%" alt="" /><figcaption><span>Figure 9:</span> How N3C Data Governance is Structured.</figcaption>
</figure>
</div>
<h4 id="covid-recovery-corps-crc">COVID Recovery Corps (CRC)</h4>
<p>The Covid Recovery Corps (CRC) is a collaborative research study by Columbia University and Sage Bionetworks, supported by the Chan Zuckerberg Initiative. The CRC aims to enroll people from the New York City metro area and have recovered from COVID-19 to partner with scientists to better understand the recovery process and long term effects from the disease.</p>
<p>Some patterns of the CRC are stable governance design patterns: it’s a research study feeding data into a club under clinical protocols and using informed consent. The study also launches with a 50,000 person Facebook community, anticipates national at-home testing, and its protocol must contemplate cities other than New York joining the protocol as new “nodes” on a clinical network.</p>
<p>Using governance structures and design patterns, less than 50 days elapsed between kickoff meetings and IRB approval for the project.</p>
<div id="fig:fig10" class="fignos">
<figure>
<img src="images/figure10.png" style="width:80.0%" alt="" /><figcaption><span>Figure 10:</span> How CRC Data Governance is Structured.</figcaption>
</figure>
</div>
<h3 id="governance-and-design-in-context">Governance and Design In Context</h3>
<p>Our approach in this Green Paper sits within a broader movement to apply user-centered design to policy. The concepts of structures and patterns will be familiar to designers. Part of our argument is that governance needs design, and that designers should embrace governance as a topic area.</p>
<p>A well-designed governance system must center empathy for those using it, and frame the design work in terms of their experience of use. While governance creation often starts with desired outcome (e.g., “more fairness” or “more ethics” or “faster outcomes” or “collaboration formed”) it is rare that those outcomes include or articulate aspects of the lived experience of technology-based biology/data science. Designing governance systems for data science, like other digital services, should begin by exploring and pinpointing the needs of the people who will use the service, and the ways the service will fit into their lives.</p>
<p>One helpful tool for user-centered design is the persona. Based on interviews with a wide variety of users to understand their views and experiences, it is possible to derive a handful of patterns in the types of users, each augmented with vivid details — a name, picture, personal history, etc — to conjure an image of the person being designed for. Personas provide user archetypes that capture the essentials and are easy to recall and reference throughout the design process. In the context of data use, personas could allow us to test a governance feature with respect to the needs of a variety of stakeholders in open science.</p>
<p>Scenarios represent another tool. Like a persona, a scenario is an artifice that is based on empirical research, analysis of patterns, and synthesis of attributes, embellished with colorful, memorable details to aid in functionality. However, instead of creating a set of hypothetical people, we are creating a set of hypothetical contexts in which they would act. Like personas, scenarios define the space of the possible, and are useful only as a set (not individually) to explore the robustness of particular options. Also, a hallmark of user-centered design is frequent, rapid, iterative prototyping, which incorporates user feedback back into the design process as often as possible. Testbeds to enable frequent, rapid, iterative prototyping and integrate user feedback are another often overlooked, but entirely necessary, component of the governance enterprise.</p>
<h2 id="conclusion">Conclusion</h2>
<p>Data use collaborations are incredibly diverse in terms of the types of artifacts being shared, the degree to which they are shared, the purposes with which they are shared, not to mention the power dynamics among collaborators. This means that we must attend to the design of the rules and incentive structures that govern human behavior as much as we attend to the design of hardware and software. While there is no one-size-fits-all solution, we can learn from observed research collaborations and reuse patterns of governance systems that worked well for those collaborations when we design new systems of governance, or transition from one system to another. The important thing is that form follows function: governance must be determined by the nature of collaboration desired. For example, large projects require governance campaigns and design, are high impact and rare, and often possess long feedback loops before value creation. Small projects require on-the-ground governance and immediate feedback loops.</p>
<p>There are limits to the structure-design pattern approach. One is that implementation is messy in reality: there will inevitably be some trial and error, as technology and governance are co-iterated in practice. Exactly how technology and governance combine in actual use will be determined when the project runs. Additionally, institutions within science have longstanding incentives to perform customized governance as a service, which can make traction for new methods difficult. Funder mandates have been quite successful in open access governance and may be necessary in data governance as well.</p>
<p>Thus by our model we do not mean there is one “right” governance design, but instead that successful designs must share two traits: <strong>they should allow collaborators to get going quickly in a legally valid structure, and they should also be flexible enough to change to meet the needs of collaborators</strong> — especially in the early stages of design or transition. Turning ad hoc processes into governance design patterns means they can be encoded in software: services and products can integrate some key patterns and variables, supporting implementation in daily practice.</p>
<p>Our structure-design pattern for data governance pairing sits inside a growing understanding of how the ways that human-centered design can help in policymaking and governance. In addition to this Green Paper, which will map out the landscape of data governance and its mixture into technical platforms, we are in early development for three major projects that take advantage of that mixture. Each draws on Sage’s goals of increasing the scope of responsible sharing, of representative data, to increase the reliability of claims that come out of data analysis.</p>
<p>Like our structure-design pattern work here, the next stage builds on the way we develop software and other products using human-centered design. The work of using open science to generate reliable claims from representative data (and of sharing both responsibly) remains laborious, manual, and time consuming. We propose to extend data governance into software testing frameworks and information architecture to address this problem.</p>
<h5 id="reliable-analytic-test-framework">Reliable Analytic Test Framework</h5>
<p>Our Reliable Analytic Environment project looks to extend the concept of automated testing frameworks from software testing to create similar functionality to evaluate the reliability of analytical outputs. Instead of looking for software bugs, we will look for reliability bugs and their impacts. These could come from small sample sizes, failure to implement known best practices, and failure to test against known benchmarks.</p>
<h5 id="representative-algorithm-test-framework">Representative Algorithm Test Framework</h5>
<p>We propose to create a testing framework for representativeness in medical research. Drawing on research into social determinants of health and the rapidly increasing public health data sources available, we can begin to craft software services that take a person’s algorithmic designs and test them against public, semi-public, and potentially even semi-private data sets to evaluate the likelihood of under-sampling bias. This will allow data scientists to examine how their code looks beyond their non-representative samples, and see if they are accidentally encoding bias in their work — and, if so, what the impacts are.</p>
<h5 id="information-architecture-of-data-governance">Information Architecture of Data Governance</h5>
<p>For areas such as informed consent, privacy policy, contract negotiation, and data licensing, simply generating and releasing tools and licenses has not been shown to have a nonlinear impact. One reason is that an enormous amount of tacit knowledge is required to use these kinds of tools appropriately, and they are often used in spaces that are distant from the data scientist (i.e., lawyers discuss these tools, not coders). Our IA work will explore how we can learn from libraries and other complex information spaces, so that we can find areas where we need to build tools. Tools could include pedagogical resources (up to and including online courses), software services, and testing environments.</p>
<h3 id="epilogue">Epilogue</h3>
<p>We envision the integrated technologies, governance structures, and design patterns we describe here as part of a fundamental infrastructure for a robust science commons. It marries the development of technologies for open science with the development of technologies for their governance. In our vision, the science commons infrastructure would connect scientific knowledge — at least in part — across domains and guide the evolution of data science across scholarly paradigms. Every scientist would be afforded access to digital services with which to contribute to and benefit from a scientific knowledge commons and be able, in turn, to manage those services legally and ethically. For assets from data to algorithms to protocols to prose to diagrams, anyone could store, transfer, compute, collaborate, publish, read, verify, and critique. This infrastructure will require the hardware and software to create virtual spaces for data sharing, as well as the governance systems to make data sharing effective, efficient, and responsible.</p>
<p>Effective governance mechanisms would incentivize high quality scientific work and protect against abuses. For example, virtual spaces could be certified for different purposes: zones of complete openness, zones of privacy, zones that are mixed, and experimental zones that allow for work beyond default rules. Across spaces with different rules, it may also make sense to allow for selective enforcement to stimulate innovative activity in virtual “bohemias.” Such spaces could benefit emergent, quasi-scientific domains (e.g. DIY biology), which need more structure and connection to established fields than they presently have, but not so much that it dulls the novel nature of the work. Like hardware and software, governance would have to be tailored and evolve over time.</p>
<p>Creating a digital infrastructure for the science commons is no small undertaking. But science has always faced a tragedy of the commons. Vannevar Bush wrote in his now infamous text, <em>Science: the Endless Frontier</em>: “there are areas of science in which the public interest is acute but which are likely to be cultivated inadequately if left without more support than will come from private sources.” For the last 75 years, the federal government has stepped in to address this market failure through the National Science Foundation, a constellation of National Laboratories, and other means. “Basic research leads to new knowledge. It provides scientific capital. It creates the fund from which the practical applications of knowledge must be drawn.”<span class="citation" data-cites="14prwQcfV"><sup><a href="#ref-14prwQcfV" role="doc-biblioref">8</a></sup></span></p>
<p>In our science-innovation mythos, the commons was understood narrowly as fundamental scientific knowledge — the “seedcorn” of innovation — and the tragedy of its underproduction was to be solved by public funding of discipline-based, investigator-driven, basic research. But the world has changed over the last 75 years. Notably, there is no longer a market failure for R&amp;D. Private funding of R&amp;D has surpassed federal funding since 1980 and that gap has been steadily growing,<span class="citation" data-cites="T5HotgQ3"><sup><a href="#ref-T5HotgQ3" role="doc-biblioref">21</a></sup></span> and “innovations often occur that do not require basic or applied research or development.”<span class="citation" data-cites="17uRi6Qj3"><sup><a href="#ref-17uRi6Qj3" role="doc-biblioref">12</a></sup></span> Despite the growth of scientific knowledge over the past decades, economic productivity has stagnated; more R&amp;D at universities does not automatically translate into more value in the economy.<span class="citation" data-cites="uLLZUEu"><sup><a href="#ref-uLLZUEu" role="doc-biblioref">4</a></sup></span> Something is missing. The scientific commons has not disappeared, but it has shifted… and so has the tragedy.</p>
<p>The old commons tragedy was the production of <em>knowledge</em>, with the focus on the product: our metaphorical seed corn. The new commons tragedy is the <em>production</em> of knowledge, with the focus on the production: our metaphorical system of agriculture. Seed corn cannot grow if the topsoil has been eroded, or if a drought has reduced the amount of available water, or if a shifting climate has increased the number or freezing degree days, or if bollweevils eat all of our seedcorn because, though monoculture, our stock was not diverse enough. If collecting more seedcorn is our only tactic, then we may go hungry when our environmental context changes. Instead, we must have a more diverse set of tactics within an overarching, robust strategy of cultivation.</p>
<p>How will the practice of science, itself, operate for the good of society in our new era? We must reconceive the scientific commons as a system and, therefore, pay more attention to what lies between individual sciences: the transdisciplinary infrastructure and boundary-spanning institutions that enable the entire system of knowledge production to function. We must build the tools and practices for integrating different communities of knowledge workers into a larger ecology of knowledge in order to make the system as a whole more productive and more resilient.</p>
<p>Good governance for everyone is where we propose to begin.</p>
<h2 id="acknowledgments">Acknowledgments</h2>
<p>We are grateful to Microsoft for supporting this paper, particularly Nicolas Schifano and Katherine Spelman.</p>
<p>This paper would not exist without the Sage Bionetworks team. Christine Suver contributed enormously to the governance structures described here, and Megan Doerr’s work on informed consent and data governance is woven throughout. Vanessa Barone’s work with Woody MacDuffie on the Privacy Toolkit contributed the key concept of design patterns. Sarah Moore and Victoria Allen’s work on the AllofUs Research Program represents an ongoing test of the models described here. And Stephen Friend was a constant early inspiration and collaborator for innovation in governance structures.</p>
<p>We would also like to thank John Chaffins, Stephanie Devaney, Jesse Dylan, Ruth Faden, Kadija Ferryman, David Fore, Nancy Kass, Bartha Knoppers, Sean McDonald, Jasmine McNealy, Michelle Meyer, Stephanie Nguyen, Adrian Thorogood, Jennifer Wagner, and Joon-Ho Yu for ongoing conversations that contributed to this paper.</p>
<p>All mistakes are ours.</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references" role="doc-bibliography">
<div id="ref-12mne7E5t">
<p>1. A data processing addendum for the CCPA? [accessed 2020 Jun 18]. <a href="https://iapp.org/news/a/a-data-processing-addendum-for-the-ccpa/">https://iapp.org/news/a/a-data-processing-addendum-for-the-ccpa/</a></p>
</div>
<div id="ref-10nEXzoVA">
<p>2. All of Us Research Program Begins Beta Testing of Data Platform | National Institutes of Health (NIH) — All of Us. [accessed 2020 Jun 18]. <a href="https://allofus.nih.gov/news-events-and-media/announcements/all-us-research-program-begins-beta-testing-data-platform">https://allofus.nih.gov/news-events-and-media/announcements/all-us-research-program-begins-beta-testing-data-platform</a></p>
</div>
<div id="ref-1HXhjeVQY">
<p>3. All of Us Research Program IRB | National Institutes of Health. [accessed 2020 Jun 18]. <a href="https://allofus.nih.gov/about/who-we-are/institutional-review-board-irb-of-all-of-us-research-program">https://allofus.nih.gov/about/who-we-are/institutional-review-board-irb-of-all-of-us-research-program</a></p>
</div>
<div id="ref-uLLZUEu">
<p>4. Arora A, Belenzon S, Patacconi A, Suh J. The Changing Structure of American Innovation: Some Cautionary Remarks for Economic Growth. National Bureau of Economic Research. 2019 Jun 28. <a href="https://doi.org/gg2jdt">https://doi.org/gg2jdt</a>. doi:<a href="https://doi.org/10.3386/w25893">10.3386/w25893</a></p>
</div>
<div id="ref-H3YsDbZv">
<p>5. Bardill J, Garrison NA. Naming Indigenous Concerns, Framing Considerations for Stored Biospecimens. The American Journal of Bioethics. 2015;15(9):73–75. <a href="https://doi.org/gg2jdp">https://doi.org/gg2jdp</a>. doi:<a href="https://doi.org/10.1080/15265161.2015.1062164">10.1080/15265161.2015.1062164</a></p>
</div>
<div id="ref-LArYGREf">
<p>6. Bot BM, Suver C, Neto EC, Kellen M, Klein A, Bare C, Doerr M, Pratap A, Wilbanks J, Dorsey ER, et al. The mPower study, Parkinson disease mobile data collected using ResearchKit. Scientific Data. 2016;3(1). <a href="https://doi.org/gg2dvj">https://doi.org/gg2dvj</a>. doi:<a href="https://doi.org/10.1038/sdata.2016.11">10.1038/sdata.2016.11</a></p>
</div>
<div id="ref-M5iKzYGg">
<p>7. Buolamwini J, Gebru T. Gender shades: Intersectional accuracy disparities in commercial gender classification. Conference on fairness, accountability and transparency. 2018 Jul 16. <a href="http://proceedings.mlr.press/v81/buolamwini18a/buolamwini18a.pdf">http://proceedings.mlr.press/v81/buolamwini18a/buolamwini18a.pdf</a></p>
</div>
<div id="ref-14prwQcfV">
<p>8. Bush V. Science the Endless Frontier. 1945 [accessed 2020 Jun 18]. <a href="https://www.nsf.gov/od/lpa/nsf50/vbush1945.htm">https://www.nsf.gov/od/lpa/nsf50/vbush1945.htm</a></p>
</div>
<div id="ref-VG5MXB5B">
<p>9. Cheah PY, Piasecki J. Data Access Committees. BMC Medical Ethics. 2020;21(1). <a href="https://doi.org/gg2d3z">https://doi.org/gg2d3z</a>. doi:<a href="https://doi.org/10.1186/s12910-020-0453-z">10.1186/s12910-020-0453-z</a></p>
</div>
<div id="ref-10J8KhM0p">
<p>10. Cheah PY, Tangseefa D, Somsaman A, Chunsuttiwat T, Nosten F, Day NPJ, Bull S, Parker M. Perceived Benefits, Harms, and Views About How to Share Data Responsibly. Journal of Empirical Research on Human Research Ethics. 2015;10(3):278–289. <a href="https://doi.org/f7pmw8">https://doi.org/f7pmw8</a>. doi:<a href="https://doi.org/10.1177/1556264615592388">10.1177/1556264615592388</a></p>
</div>
<div id="ref-32SWh1ey">
<p>11. Collins FS. The Human Genome Project: Lessons from Large-Scale Biology. Science. 2003;300(5617):286–290. <a href="https://doi.org/d2zg2r">https://doi.org/d2zg2r</a>. doi:<a href="https://doi.org/10.1126/science.1084564">10.1126/science.1084564</a></p>
</div>
<div id="ref-17uRi6Qj3">
<p>12. Congressional Research Service. Industrial Competitiveness and Technological Advancement: Debate Over Government Policy. 2012. <a href="https://www.everycrsreport.com/files/20121203_RL33528_ebc4cd402342b9bbf91bebdb37bbe6bdc06e57ae.pdf">https://www.everycrsreport.com/files/20121203_RL33528_ebc4cd402342b9bbf91bebdb37bbe6bdc06e57ae.pdf</a></p>
</div>
<div id="ref-bQ5U0G1W">
<p>13. Creative Commons — CC0 1.0 Universal. [accessed 2020 Jun 18]. <a href="https://creativecommons.org/publicdomain/zero/1.0/">https://creativecommons.org/publicdomain/zero/1.0/</a></p>
</div>
<div id="ref-xQxy7NV0">
<p>14. Dataset Downloads | CMS. [accessed 2020 Jun 18]. <a href="https://www.cms.gov/OpenPayments/Explore-the-Data/Dataset-Downloads">https://www.cms.gov/OpenPayments/Explore-the-Data/Dataset-Downloads</a></p>
</div>
<div id="ref-80a55iFY">
<p>15. Data Use Policies – All of Us Research Hub. [accessed 2020 Jun 18]. <a href="https://www.researchallofus.org/data-use-policies/">https://www.researchallofus.org/data-use-policies/</a></p>
</div>
<div id="ref-wqupV8k1">
<p>16. Decker R, Wimsatt L, Trice A, Konstan J. A profile of federal-grant administrative burden among Federal Demonstration Partnership faculty. 2007. <a href="https://sites.nationalacademies.org/cs/groups/pgasite/documents/webpage/pga_054586.pdf">https://sites.nationalacademies.org/cs/groups/pgasite/documents/webpage/pga_054586.pdf</a></p>
</div>
<div id="ref-14K4iV022">
<p>17. De Identified Information Sample Clauses. [accessed 2020 Jun 18]. <a href="https://www.lawinsider.com/clause/de-identified-information">https://www.lawinsider.com/clause/de-identified-information</a></p>
</div>
<div id="ref-1DGkp34ZB">
<p>18. Dyke SOM, Philippakis AA, Rambla De Argila J, Paltoo DN, Luetkemeier ES, Knoppers BM, Brookes AJ, Spalding JD, Thompson M, Roos M, et al. Consent Codes: Upholding Standard Data Use Conditions. PLOS Genetics. 2016;12(1):e1005772. <a href="https://doi.org/gg2d32">https://doi.org/gg2d32</a>. doi:<a href="https://doi.org/10.1371/journal.pgen.1005772">10.1371/journal.pgen.1005772</a></p>
</div>
<div id="ref-11WcMMgG1">
<p>19. Elements of Informed Consent. Sage Bionetworks. [accessed 2020 Jun 18]. <a href="https://sagebionetworks.org/tools_resources/elements-of-informed-consent/">https://sagebionetworks.org/tools_resources/elements-of-informed-consent/</a></p>
</div>
<div id="ref-EIBAJZJB">
<p>20. Ferryman K, Pitcan M. Fairness in precision medicine. Data &amp; Society. 2018 Feb. <a href="https://datasociety.net/wp-content/uploads/2018/02/DataSociety_Fairness_In_Precision_Medicine_Feb2018.pdf">https://datasociety.net/wp-content/uploads/2018/02/DataSociety_Fairness_In_Precision_Medicine_Feb2018.pdf</a></p>
</div>
<div id="ref-T5HotgQ3">
<p>21. Figure 4-1 - S&amp;E Indicators 2018 | NSF - National Science Foundation. [accessed 2020 Jun 18]. <a href="https://www.nsf.gov/statistics/2018/nsb20181/figures/fig04-01">https://www.nsf.gov/statistics/2018/nsb20181/figures/fig04-01</a></p>
</div>
<div id="ref-ENjTbvYJ">
<p>22. FormSwift: Create Legal Documents. [accessed 2020 Jun 18]. <a href="https://formswift.com/">https://formswift.com/</a></p>
</div>
<div id="ref-isF1I1NM">
<p>23. Fuller E. Dr. Schneider’s FDP Presentation. 2020 Apr 7. <a href="http://thefdp.org/default/assets/File/Presentations/Schneider%20FDP%20FWS3%20Results%20Plenary%20Jan19.pdf">http://thefdp.org/default/assets/File/Presentations/Schneider%20FDP%20FWS3%20Results%20Plenary%20Jan19.pdf</a></p>
</div>
<div id="ref-ZvwpEFTA">
<p>24. Garrison NA. Genomic Justice for Native Americans. Science, Technology, &amp; Human Values. 2012;38(2):201–223. <a href="https://doi.org/gg2dvk">https://doi.org/gg2dvk</a>. doi:<a href="https://doi.org/10.1177/0162243912470009">10.1177/0162243912470009</a></p>
</div>
<div id="ref-1As3pvk5w">
<p>25. Grayson S, Suver C, Wilbanks J, Doerr M. Open Data Sharing in the 21st Century: Sage Bionetworks’ Qualified Research Program and Its Application in mHealth Data Release. SSRN Electronic Journal. 2019. <a href="https://doi.org/gg2jds">https://doi.org/gg2jds</a>. doi:<a href="https://doi.org/10.2139/ssrn.3502410">10.2139/ssrn.3502410</a></p>
</div>
<div id="ref-12HajymAr">
<p>26. Greshake Tzovaras B, Angrist M, Arvai K, Dulaney M, Estrada-Galiñanes V, Gunderson B, Head T, Lewis D, Nov O, Shaer O, et al. Open Humans: A platform for participant-centered research and personal data exploration. GigaScience. 2019;8(6). <a href="https://doi.org/gg2jdq">https://doi.org/gg2jdq</a>. doi:<a href="https://doi.org/10.1093/gigascience/giz076">10.1093/gigascience/giz076</a></p>
</div>
<div id="ref-KLKqRnt">
<p>27. Guinney J, Dienstmann R, Wang X, de Reyniès A, Schlicker A, Soneson C, Marisa L, Roepman P, Nyamundanda G, Angelino P, et al. The consensus molecular subtypes of colorectal cancer. Nature Medicine. 2015;21(11):1350–1356. <a href="https://doi.org/f7xmph">https://doi.org/f7xmph</a>. doi:<a href="https://doi.org/10.1038/nm.3967">10.1038/nm.3967</a></p>
</div>
<div id="ref-nw5braAS">
<p>28. Guinney J, Saez-Rodriguez J. Alternative models for sharing confidential biomedical data. Nature Biotechnology. 2018;36(5):391–392. <a href="https://doi.org/gg2dvh">https://doi.org/gg2dvh</a>. doi:<a href="https://doi.org/10.1038/nbt.4128">10.1038/nbt.4128</a></p>
</div>
<div id="ref-j7nvT8JA">
<p>29. Health Care Systems Network. (n.d.). [accessed 2020 Jun 18]. <a href="http://www.hcsrn.org/en/Tools%20&amp;%20Materials/GrantsContracting/HCSRN_DUAToolkit.pdf">http://www.hcsrn.org/en/Tools%20&amp;%20Materials/GrantsContracting/HCSRN_DUAToolkit.pdf</a></p>
</div>
<div id="ref-AnAX84nk">
<p>30. Home - The Federal Demonstration Partnership. [accessed 2020 Jun 18]. <a href="https://thefdp.org/default/">https://thefdp.org/default/</a></p>
</div>
<div id="ref-11z0HOBna">
<p>31. Knorr-Cetina K. Epistemic cultures: how the sciences make knowledge. Cambridge, Mass: Harvard University Press; 1999.</p>
</div>
<div id="ref-MmeIANuu">
<p>32. Lane J, Shipp S. Using a Remote Access Data Enclave for Data Dissemination. International Journal of Digital Curation. 2008;2(1):128–134. <a href="https://doi.org/dfkk4c">https://doi.org/dfkk4c</a>. doi:<a href="https://doi.org/10.2218/ijdc.v2i1.20">10.2218/ijdc.v2i1.20</a></p>
</div>
<div id="ref-G3NU3hO7">
<p>33. LegalZoom: Start a Business, Protect Your Family. LegalZoom. [accessed 2020 Jun 18]. <a href="https://www.legalzoom.com">https://www.legalzoom.com</a></p>
</div>
<div id="ref-t5HXYtyn">
<p>34. Material Transfer Agreements (MTAs) for Technology | AUTM. [accessed 2020 Jun 18]. <a href="https://autm.net/surveys-and-tools/agreements/material-transfer-agreements/mta-toolkit/uniform-biological-material-transfer-agreement">https://autm.net/surveys-and-tools/agreements/material-transfer-agreements/mta-toolkit/uniform-biological-material-transfer-agreement</a></p>
</div>
<div id="ref-1DMYQt6hx">
<p>35. Mazur DJ. Evaluating the science and ethics of research on humans: a guide for IRB members. Baltimore: Johns Hopkins University Press; 2007.</p>
</div>
<div id="ref-TUd0l6ty">
<p>36. McDonald S. Reclaiming Data Trusts. Centre for International Governance Innovation. [accessed 2020 Jun 18]. <a href="https://www.cigionline.org/articles/reclaiming-data-trusts">https://www.cigionline.org/articles/reclaiming-data-trusts</a></p>
</div>
<div id="ref-tPpeCQYm">
<p>37. Michel KK. IDENTIFICATION OF PRIMARY TIME DRIVERS IN COMPLIANCE FOR NON-MEDICAL STEM RESEARCH, AND BEST PRACTICES FOR REDUCING TIME INVESTMENT BY FACULTY [Thesis]. Johns Hopkins University; 2019 [accessed 2020 Jun 18]. <a href="http://dspace-prod.mse.jhu.edu:8080/handle/1774.2/62063">http://dspace-prod.mse.jhu.edu:8080/handle/1774.2/62063</a></p>
</div>
<div id="ref-zCTNJz8u">
<p>38. microsoft/Open-Use-of-Data-Agreement. Microsoft; 2020 [accessed 2020 Jun 18]. <a href="https://github.com/microsoft/Open-Use-of-Data-Agreement">https://github.com/microsoft/Open-Use-of-Data-Agreement</a></p>
</div>
<div id="ref-BtzWfCJm">
<p>39. National Center for Biotechnology Information. (n.d.). DbGaP Request Procedures to Access Individual-Level Data. 2011 Mar 23. <a href="https://dbgap.ncbi.nlm.nih.gov/aa/dbgap_request_process.pdf">https://dbgap.ncbi.nlm.nih.gov/aa/dbgap_request_process.pdf</a></p>
</div>
<div id="ref-5plIyD5u">
<p>40. Nugent T, Upton D, Cimpoesu M. Improving data transparency in clinical trials using blockchain smart contracts. F1000Research. 2016;5:2541. <a href="https://doi.org/ggzsn4">https://doi.org/ggzsn4</a>. doi:<a href="https://doi.org/10.12688/f1000research.9756.1">10.12688/f1000research.9756.1</a></p>
</div>
<div id="ref-1BpKAGz3u">
<p>41. Office for Civil Rights (OCR): Business Associate Contracts. HHS.gov. 2008 May 21 [accessed 2020 Jun 18]. <a href="https://www.hhs.gov/hipaa/for-professionals/covered-entities/sample-business-associate-agreement-provisions/index.html">https://www.hhs.gov/hipaa/for-professionals/covered-entities/sample-business-associate-agreement-provisions/index.html</a></p>
</div>
<div id="ref-1A5Jsy5oq">
<p>42. Open Data Commons Open Database License (ODbL). Open Data Commons. 2009 [accessed 2020 Jun 18]. <a href="https://opendatacommons.org/licenses/odbl/">https://opendatacommons.org/licenses/odbl/</a></p>
</div>
<div id="ref-f8vwfMHl">
<p>43. Open Data Commons Public Domain Dedication and License (PDDL). Open Data Commons. 2009 [accessed 2020 Jun 18]. <a href="https://opendatacommons.org/licenses/pddl/">https://opendatacommons.org/licenses/pddl/</a></p>
</div>
<div id="ref-1F7XRqf7Z">
<p>44. OpenDP. GitHub. [accessed 2020 Jun 18]. <a href="https://github.com/opendifferentialprivacy">https://github.com/opendifferentialprivacy</a></p>
</div>
<div id="ref-abUbT01W">
<p>45. Ostrom E. Beyond Markets and States: Polycentric Governance of Complex Economic Systems. American Economic Review. 2010;100(3):641–672. <a href="https://doi.org/dhj38d">https://doi.org/dhj38d</a>. doi:<a href="https://doi.org/10.1257/aer.100.3.641">10.1257/aer.100.3.641</a></p>
</div>
<div id="ref-6j990tze">
<p>46. Pasquetto IV, Randles BM, Borgman CL. On the Reuse of Scientific Data. Data Science Journal. 2017;16. <a href="https://doi.org/gf87pf">https://doi.org/gf87pf</a>. doi:<a href="https://doi.org/10.5334/dsj-2017-008">10.5334/dsj-2017-008</a></p>
</div>
<div id="ref-hNKKknXk">
<p>47. PRM | CDISC. [accessed 2020 Jun 18]. <a href="https://www.cdisc.org/standards/foundational/protocol">https://www.cdisc.org/standards/foundational/protocol</a></p>
</div>
<div id="ref-CxfmIM8F">
<p>48. Protocol Templates for Clinical Trials | grants.nih.gov. [accessed 2020 Jun 18]. <a href="https://grants.nih.gov/policy/clinical-trials/protocol-template.htm">https://grants.nih.gov/policy/clinical-trials/protocol-template.htm</a></p>
</div>
<div id="ref-vzr6hitM">
<p>49. Ross MW, Iguchi MY, Panicker S. Ethical aspects of data sharing and research participant protections. American Psychologist. 2018;73(2):138–145. <a href="https://doi.org/gc4823">https://doi.org/gc4823</a>. doi:<a href="https://doi.org/10.1037/amp0000240">10.1037/amp0000240</a></p>
</div>
<div id="ref-e167exJ6">
<p>50. Sample Addendum Addressing Article 28 GDPR and Incorporating Standard Contractual Clauses for Controller to Processor Transfers of Personal Data. [accessed 2020 Jun 18]. <a href="https://iapp.org/resources/article/sample-addendum-addressing-article-28-gdpr-and-incorporating-standard-contractual-clauses-for-controller-to-processor-transfers-of-personal-data/">https://iapp.org/resources/article/sample-addendum-addressing-article-28-gdpr-and-incorporating-standard-contractual-clauses-for-controller-to-processor-transfers-of-personal-data/</a></p>
</div>
<div id="ref-odIgeu98">
<p>51. Schaffter T, Buist DSM, Lee CI, Nikulin Y, Ribli D, Guan Y, Lotter W, Jie Z, Du H, Wang S, et al. Evaluation of Combined Artificial Intelligence and Radiologist Assessment to Interpret Screening Mammograms. JAMA Network Open. 2020;3(3):e200265. <a href="https://doi.org/gg2dvg">https://doi.org/gg2dvg</a>. doi:<a href="https://doi.org/10.1001/jamanetworkopen.2020.0265">10.1001/jamanetworkopen.2020.0265</a></p>
</div>
<div id="ref-TQlIEC6o">
<p>52. Schwarte A, Haase P, Hose K, Schenkel R, Schmidt M. FedX: Optimization Techniques for Federated Query Processing on Linked Data. In: Lecture Notes in Computer Science. Springer Berlin Heidelberg; 2011. <a href="https://doi.org/bgnp5d">https://doi.org/bgnp5d</a>. doi:<a href="https://doi.org/10.1007/978-3-642-25073-6_38">10.1007/978-3-642-25073-6_38</a></p>
</div>
<div id="ref-dIYXacYW">
<p>53. Seyednasrollah F, Koestler DC, Wang T, Piccolo SR, Vega R, Greiner R, Fuchs C, Gofer E, Kumar L, Wolfinger RD, et al. A DREAM Challenge to Build Prediction Models for Short-Term Discontinuation of Docetaxel in Metastatic Castration-Resistant Prostate Cancer. JCO Clinical Cancer Informatics. 2017;(1):1–15. <a href="https://doi.org/gg2jdr">https://doi.org/gg2jdr</a>. doi:<a href="https://doi.org/10.1200/cci.17.00018">10.1200/cci.17.00018</a></p>
</div>
<div id="ref-116CDX0ks">
<p>54. Share your work. Creative Commons. [accessed 2020 Jun 18]. <a href="https://creativecommons.org/share-your-work/">https://creativecommons.org/share-your-work/</a></p>
</div>
<div id="ref-wthFx3D2">
<p>55. Standard Contractual Clauses (SCC). European Commission - European Commission. [accessed 2020 Jun 18]. <a href="https://ec.europa.eu/info/law/law-topic/data-protection/international-dimension-data-protection/standard-contractual-clauses-scc_en">https://ec.europa.eu/info/law/law-topic/data-protection/international-dimension-data-protection/standard-contractual-clauses-scc_en</a></p>
</div>
<div id="ref-H5KgzA16">
<p>56. Template Studio - Accord Project. [accessed 2020 Jun 18]. <a href="https://www.accordproject.org/projects/template-studio/">https://www.accordproject.org/projects/template-studio/</a></p>
</div>
<div id="ref-1GcAkA1FX">
<p>57. Vaidhyanathan S. Antisocial media: how facebook disconnects US and undermines democracy. New York, NY, United States of America: Oxford University Press; 2018.</p>
</div>
<div id="ref-H7PziDQ7">
<p>58. Walonoski J, Kramer M, Nichols J, Quina A, Moesel C, Hall D, Duffett C, Dube K, Gallagher T, McLachlan S. Synthea: An approach, method, and software mechanism for generating synthetic patients and the synthetic electronic health care record. Journal of the American Medical Informatics Association. 2018;25(3):230–238. <a href="https://doi.org/gb4mkg">https://doi.org/gb4mkg</a>. doi:<a href="https://doi.org/10.1093/jamia/ocx079">10.1093/jamia/ocx079</a></p>
</div>
<div id="ref-1mFzspuo">
<p>59. Wellcome Sanger Institute: Catalogue of Somatic Mutations in Cancer Data Downloads. [accessed 2020 Jun 18]. <a href="https://cancer.sanger.ac.uk/cosmic/download?genome=37">https://cancer.sanger.ac.uk/cosmic/download?genome=37</a></p>
</div>
<div id="ref-1Cx91jrTR">
<p>60. What is a Privacy Toolkit? | Design System Template. [accessed 2020 Jun 18]. <a href="https://designmanual.sagebionetworks.org/privacy_toolkit.html">https://designmanual.sagebionetworks.org/privacy_toolkit.html</a></p>
</div>
<div id="ref-otY3SefD">
<p>61. Wilbanks J, Friend SH. First, design for data sharing. Nature Biotechnology. 2016;34(4):377–379. <a href="https://doi.org/gbrb37">https://doi.org/gbrb37</a>. doi:<a href="https://doi.org/10.1038/nbt.3516">10.1038/nbt.3516</a></p>
</div>
<div id="ref-Piezz6R5">
<p>62. Wilkins CH, Schindler SE, Morris JC. Addressing Health Disparities Among Minority Populations. JAMA Neurology. 2020 Jun 15. <a href="https://doi.org/gg2jdn">https://doi.org/gg2jdn</a>. doi:<a href="https://doi.org/10.1001/jamaneurol.2020.1614">10.1001/jamaneurol.2020.1614</a></p>
</div>
<div id="ref-6DjakjNS">
<p>63. Wilkinson MD, Dumontier M, Aalbersberg IJ, Appleton G, Axton M, Baak A, Blomberg N, Boiten J-W, da Silva Santos LB, Bourne PE, et al. The FAIR Guiding Principles for scientific data management and stewardship. Scientific Data. 2016;3(1). <a href="https://doi.org/bdd4">https://doi.org/bdd4</a>. doi:<a href="https://doi.org/10.1038/sdata.2016.18">10.1038/sdata.2016.18</a></p>
</div>
<div id="ref-1EZbXJTg3">
<p>64. WIRB Home. [accessed 2020 Jun 18]. <a href="http://www.wirb.com/Pages/Default.aspx">http://www.wirb.com/Pages/Default.aspx</a></p>
</div>
</div>
<!-- default theme -->

<style>
    /* import google fonts */
    @import url("http://fonts.googleapis.com/css?family=Lato");
    @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

    /* -------------------------------------------------- */
    /* global */
    /* -------------------------------------------------- */

    /* all elements */
    * {
        /* force sans-serif font unless specified otherwise */
        font-family: "Lato", sans-serif;

        /* prevent text inflation on some mobile browsers */
        -webkit-text-size-adjust: none !important;
        -moz-text-size-adjust: none !important;
        -o-text-size-adjust: none !important;
        text-size-adjust: none !important;
    }

    @media only screen {
        /* "page" element */
        body {
            position: relative;
            box-sizing: border-box;
            font-size: 12pt;
            line-height: 1.5;
            max-width: 8.5in;
            margin: 20px auto;
            padding: 40px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* "page" element */
        body {
            padding: 20px;
            margin: 0;
            border-radius: 0;
            border: none;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
            background: none;
        }
    }

    /* -------------------------------------------------- */
    /* headings */
    /* -------------------------------------------------- */

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        margin: 20px 0;
        padding: 0;
        font-weight: bold;
    }

    /* biggest heading */
    h1 {
        margin: 40px 0;
        text-align: center;
    }

    /* second biggest heading */
    h2 {
        margin-top: 30px;
        padding-bottom: 5px;
        border-bottom: solid 1px #bdbdbd;
    }

    /* heading font sizes */
    h1 {
        font-size: 2em;
    }
    h2 {
        font-size: 1.5em;
    }
    h3{
        font-size: 1.35em;
    }
    h4 {
        font-size: 1.25em;
    }
    h5 {
        font-size: 1.15em;
    }
    h6 {
        font-size: 1em;
    }

    /* -------------------------------------------------- */
    /* manuscript header */
    /* -------------------------------------------------- */

    /* manuscript title */
    header > h1 {
        margin: 0;
    }

    /* manuscript title caption text (ie "automatically generated on") */
    header + p {
        text-align: center;
        margin-top: 10px;
    }

    /* -------------------------------------------------- */
    /* text elements */
    /* -------------------------------------------------- */

    /* links */
    a {
        color: #2196f3;
        overflow-wrap: break-word;
    }

    /* normal links (not empty, not button link, not syntax highlighting link) */
    a:not(:empty):not(.button):not(.sourceLine) {
        padding-left: 1px;
        padding-right: 1px;
    }

    /* superscripts and subscripts */
    sub,
    sup {
        /* prevent from affecting line height */
        line-height: 0;
    }

    /* unordered and ordered lists*/
    ul,
    ol {
        padding-left: 20px;
    }

    /* class for styling text semibold */
    .semibold {
        font-weight: 600;
    }

    /* class for styling elements horizontally left aligned */
    .left {
        display: block;
        text-align: left;
        margin-left: auto;
        margin-right: 0;
        justify-content: left;
    }

    /* class for styling elements horizontally centered */
    .center {
        display: block;
        text-align: center;
        margin-left: auto;
        margin-right: auto;
        justify-content: center;
    }

    /* class for styling elements horizontally right aligned */
    .right {
        display: block;
        text-align: right;
        margin-left: 0;
        margin-right: auto;
        justify-content: right;
    }

    /* -------------------------------------------------- */
    /* section elements */
    /* -------------------------------------------------- */

    /* horizontal divider line */
    hr {
        border: none;
        height: 1px;
        background: #bdbdbd;
    }

    /* paragraphs, horizontal dividers, figures, tables, code */
    p,
    hr,
    figure,
    table,
    pre {
        /* treat all as "paragraphs", with consistent vertical margins */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* figures */
    /* -------------------------------------------------- */

    /* figure */
    figure {
        max-width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure caption */
    figcaption {
        padding: 0;
        padding-top: 10px;
    }

    /* figure image element */
    figure img {
        max-width: 100%;
        display: block;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure auto-number */
    img + figcaption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* tables */
    /* -------------------------------------------------- */

    /* table */
    table {
        border-collapse: collapse;
        border-spacing: 0;
        width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* table cells */
    th,
    td {
        border: solid 1px #bdbdbd;
        padding: 10px;
        /* squash table if too wide for page by forcing line breaks */
        overflow-wrap: break-word;
        word-break: break-word;
    }

    /* header row and even rows */
    th,
    tr:nth-child(2n) {
        background-color: #fafafa;
    }

    /* odd rows */
    tr:nth-child(2n + 1) {
        background-color: #ffffff;
    }

    /* table caption */
    caption {
        text-align: left;
        padding: 0;
        padding-bottom: 10px;
    }

    /* table auto-number */
    table > caption > span:first-of-type,
    div.table_wrapper > table > caption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* code */
    /* -------------------------------------------------- */

    /* multi-line code block */
    pre {
        padding: 10px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
        break-inside: avoid;
        text-align: left;
    }

    /* inline code, ie code within normal text */
    :not(pre) > code {
        padding: 0 4px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
    }

    /* code text */
    /* apply all children, to reach syntax highlighting sub-elements */
    code,
    code * {
        /* force monospace font */
        font-family: "Source Code Pro", "Courier New", monospace;
    }

    /* -------------------------------------------------- */
    /* quotes */
    /* -------------------------------------------------- */

    /* quoted text */
    blockquote {
        margin: 0;
        padding: 0;
        border-left: 4px solid #bdbdbd;
        padding-left: 16px;
        break-inside: avoid;
    }

    /* -------------------------------------------------- */
    /* banners */
    /* -------------------------------------------------- */

    /* info banners */
    .banner {
        box-sizing: border-box;
        display: block;
        position: relative;
        width: 100%;
        margin-top: 20px;
        margin-bottom: 20px;
        padding: 20px;
        text-align: center;
    }

    /* paragraph in banner */
    .banner > p {
        margin: 0;
    }

    /* -------------------------------------------------- */
    /* highlight colors */
    /* -------------------------------------------------- */

    .white {
        background: #ffffff;
    }
    .lightgrey {
        background: #eeeeee;
    }
    .grey {
        background: #757575;
    }
    .darkgrey {
        background: #424242;
    }
    .black {
        background: #000000;
    }
    .lightred {
        background: #ffcdd2;
    }
    .lightyellow {
        background: #ffecb3;
    }
    .lightgreen {
        background: #dcedc8;
    }
    .lightblue {
        background: #e3f2fd;
    }
    .lightpurple {
        background: #f3e5f5;
    }
    .red {
        background: #f44336;
    }
    .orange {
        background: #ff9800;
    }
    .yellow {
        background: #ffeb3b;
    }
    .green {
        background: #4caf50;
    }
    .blue {
        background: #2196f3;
    }
    .purple {
        background: #9c27b0;
    }
    .white,
    .lightgrey,
    .lightred,
    .lightyellow,
    .lightgreen,
    .lightblue,
    .lightpurple,
    .orange,
    .yellow,
    .white a,
    .lightgrey a,
    .lightred a,
    .lightyellow a,
    .lightgreen a,
    .lightblue a,
    .lightpurple a,
    .orange a,
    .yellow a {
        color: #000000;
    }
    .grey,
    .darkgrey,
    .black,
    .red,
    .green,
    .blue,
    .purple,
    .grey a,
    .darkgrey a,
    .black a,
    .red a,
    .green a,
    .blue a,
    .purple a {
        color: #ffffff;
    }

    /* -------------------------------------------------- */
    /* buttons */
    /* -------------------------------------------------- */

    /* class for styling links like buttons */
    .button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        margin: 5px;
        padding: 10px 20px;
        font-size: 0.75em;
        font-weight: 600;
        text-transform: uppercase;
        text-decoration: none;
        letter-spacing: 1px;
        background: none;
        color: #2196f3;
        border: solid 1px #bdbdbd;
        border-radius: 5px;
    }

    /* buttons when hovered */
    .button:hover:not([disabled]),
    .icon_button:hover:not([disabled]) {
        cursor: pointer;
        background: #f5f5f5;
    }

    /* buttons when disabled */
    .button[disabled],
    .icon_button[disabled] {
        opacity: 0.35;
        pointer-events: none;
    }

    /* class for styling buttons containg only single icon */
    .icon_button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        text-decoration: none;
        margin: 0;
        padding: 0;
        background: none;
        border-radius: 5px;
        border: none;
        width: 20px;
        height: 20px;
        min-width: 20px;
        min-height: 20px;
    }

    /* icon button inner svg image */
    .icon_button > svg {
        height: 16px;
    }

    /* -------------------------------------------------- */
    /* icons */
    /* -------------------------------------------------- */

    /* class for styling icons inline with text */
    .inline_icon {
        height: 1em;
        position: relative;
        top: 0.125em;
    }

    /* -------------------------------------------------- */
    /* print control */
    /* -------------------------------------------------- */

    @media print {
        @page {
            /* suggested printing margin */
            margin: 0.5in;
        }

        /* document and "page" elements */
        html, body {
            margin: 0;
            padding: 0;
            width: 100%;
            height: 100%;
        }

        /* "page" element */
        body {
            font-size: 11pt !important;
            line-height: 1.35;
        }

        /* all headings */
        h1,
        h2,
        h3,
        h4,
        h5,
        h6 {
            margin: 15px 0;
        }

        /* figures and tables */
        figure, table {
            font-size: 0.85em;
        }

        /* table cells */
        th,
        td {
            padding: 5px;
        }

        /* shrink font awesome icons */
        i.fas,
        i.fab,
        i.far,
        i.fal {
            transform: scale(0.85);
        }

        /* decrease banner margins */
        .banner {
            margin-top: 15px;
            margin-bottom: 15px;
            padding: 15px;
        }

        /* class for centering an element vertically on its own page */
        .page_center {
            margin: auto;
            width: 100%;
            height: 100%;
            display: flex;
            align-items: center;
            vertical-align: middle;
            break-before: page;
            break-after: page;
        }

        /* always insert a page break before the element */
        .page_break_before {
            break-before: page;
        }

        /* always insert a page break after the element */
        .page_break_after {
            break-after: page;
        }

        /* avoid page break before the element */
        .page_break_before_avoid {
            break-before: avoid;
        }

        /* avoid page break after the element */
        .page_break_after_avoid {
            break-after: avoid;
        }

        /* avoid page break inside the element */
        .page_break_inside_avoid {
            break-inside: avoid;
        }
    }

    /* -------------------------------------------------- */
    /* override pandoc css quirks */
    /* -------------------------------------------------- */

    .sourceCode {
        /* prevent unsightly overflow in wide code blocks */
        overflow: auto !important;
    }

    div.sourceCode {
        /* prevent background fill on top-most code block  container */
        background: none !important;
    }

    .sourceCode * {
        /* force consistent line spacing */
        line-height: 1.5 !important;
    }

    div.sourceCode {
        /* style code block margins same as <pre> element */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* tablenos */
    /* -------------------------------------------------- */

    /* tablenos wrapper */
    .tablenos {
        /* show scrollbar on tables if necessary to prevent overflow */
        width: 100%;
        margin: 20px 0;
    }

    .tablenos > table {
        /* move margins from table to table_wrapper to allow margin collapsing */
        margin: 0;
    }

    @media only screen {
        /* tablenos wrapper */
        .tablenos {
            /* show scrollbar on tables if necessary to prevent overflow */
            overflow-x: auto !important;
        }

        .tablenos th,
        .tablenos td {
            overflow-wrap: unset !important;
            word-break: unset !important;
        }

        /* table in wrapper */
        .tablenos table,
        .tablenos table * {
            /* don't break table words */
            overflow-wrap: normal !important;
        }
    }

    /* -------------------------------------------------- */
    /* mathjax */
    /* -------------------------------------------------- */

    /* mathjax containers */
    .math.display > span:not(.MathJax_Preview) {
        /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
        display: flex !important;
        overflow-x: auto !important;
        overflow-y: hidden !important;
        justify-content: center;
        align-items: center;
        margin: 0 !important;
    }

    /* right click menu */
    .MathJax_Menu {
        border-radius: 5px !important;
        border: solid 1px #bdbdbd !important;
        box-shadow: none !important;
    }

    /* equation auto-number */
    span[id^="eq:"] > span.math.display + span {
        font-weight: 600;
    }

    /* equation */
    span[id^="eq:"] > span.math.display > span {
        /* nudge to make room for equation auto-number and anchor */
        margin-right: 60px !important;
    }

    /* -------------------------------------------------- */
    /* anchors plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anchor button */
        .anchor {
            opacity: 0;
            margin-left: 5px;
        }

        /* anchor buttons within <h2>'s */
        h2 .anchor {
            margin-left: 10px;
        }

        /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
        *:hover > .anchor,
        .anchor:hover,
        .anchor:focus {
            opacity: 1;
        }

        /* anchor button when hovered */
        .anchor:hover {
            cursor: pointer;
        }
    }

    /* always show anchor button on devices with no mouse/hover ability */
    @media (hover: none) {
        .anchor {
            opacity: 1;
        }
    }

    /* always hide anchor button on print */
    @media only print {
        .anchor {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* accordion plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* accordion arrow button */
        .accordion_arrow {
            margin-right: 10px;
        }

        /* arrow icon when <h2> data-collapsed attribute true */
        h2[data-collapsed="true"] > .accordion_arrow > svg {
            transform: rotate(-90deg);
        }

        /* all elements (except <h2>'s) when data-collapsed attribute true */
        *:not(h2)[data-collapsed="true"] {
            display: none;
        }

        /* accordion arrow button when hovered and <h2>'s when hovered */
        .accordion_arrow:hover,
        h2[data-collapsed="true"]:hover,
        h2[data-collapsed="false"]:hover {
            cursor: pointer;
        }
    }

    /* always hide accordion arrow button on print */
    @media only print {
        .accordion_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* tooltips plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* tooltip container */
        #tooltip {
            position: absolute;
            width: 50%;
            min-width: 240px;
            max-width: 75%;
            z-index: 1;
        }

        /* tooltip content */
        #tooltip_content {
            margin-bottom: 5px;
            padding: 20px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
            overflow-wrap: break-word;
        }

        /* tooltip copy of paragraphs and figures */
        #tooltip_content > p,
        #tooltip_content > figure {
            margin: 0;
            max-height: 320px;
            overflow-y: auto;
        }

        /* tooltip copy of <img> */
        #tooltip_content > figure > img {
            max-height: 260px;
        }

        /* navigation bar */
        #tooltip_nav_bar {
            margin-top: 10px;
            text-align: center;
        }

        /* navigation bar previous/next buton */
        #tooltip_nav_bar > .icon_button {
            position: relative;
            top: 3px;
        }

        /* navigation bar previous button */
        #tooltip_nav_bar > .icon_button:first-of-type {
            margin-right: 5px;
        }

        /* navigation bar next button */
        #tooltip_nav_bar > .icon_button:last-of-type {
            margin-left: 5px;
        }
    }

    /* always hide tooltip on print */
    @media only print {
        #tooltip {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* jump to first plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* jump button */
        .jump_arrow {
            position: relative;
            top: 0.125em;
            margin-right: 5px;
        }
    }

    /* always hide jump button on print */
    @media only print {
        .jump_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* link highlight plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anything with data-highlighted attribute true */
        [data-highlighted="true"] {
            background: #ffeb3b;
        }

        /* anything with data-selected attribute true */
        [data-selected="true"] {
            background: #ff8a65 !important;
        }

        /* animation definition for glow */
        @keyframes highlight_glow {
            0% {
                background: none;
            }
            10% {
                background: #bbdefb;
            }
            100% {
                background: none;
            }
        }

        /* anything with data-glow attribute true */
        [data-glow="true"] {
            animation: highlight_glow 2s;
        }
    }

    /* -------------------------------------------------- */
    /* table of contents plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* toc panel */
        #toc_panel {
            box-sizing: border-box;
            position: fixed;
            top: 0;
            left: 0;
            background: #ffffff;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            z-index: 2;
        }

        /* toc panel when closed */
        #toc_panel[data-open="false"] {
            min-width: 60px;
            width: 60px;
            height: 60px;
            border-right: solid 1px #bdbdbd;
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc panel when open */
        #toc_panel[data-open="true"] {
            min-width: 260px;
            max-width: 480px;
            /* keep panel edge consistent distance away from "page" edge */
            width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
            bottom: 0;
            border-right: solid 1px #bdbdbd;
        }

        /* toc panel header */
        #toc_header {
            box-sizing: border-box;
            display: flex;
            flex-direction: row;
            align-items: center;
            height: 60px;
            margin: 0;
            padding: 20px;
        }

        /* toc panel header when hovered */
        #toc_header:hover {
            cursor: pointer;
        }

        /* toc panel header when panel open */
        #toc_panel[data-open="true"] > #toc_header {
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc open/close header button */
        #toc_button {
            margin-right: 20px;
        }

        /* hide toc list and header text when closed */
        #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
        #toc_panel[data-open="false"] > #toc_list {
            display: none;
        }

        /* toc list of entries */
        #toc_list {
            box-sizing: border-box;
            width: 100%;
            padding: 20px;
            position: absolute;
            top: calc(60px + 1px);
            bottom: 0;
            overflow: auto;
        }

        /* toc entry, link to section in document */
        .toc_link {
            display: block;
            padding: 5px;
            position: relative;
            font-weight: 600;
            text-decoration: none;
        }

        /* toc entry when hovered or when "viewed" */
        .toc_link:hover,
        .toc_link[data-viewing="true"] {
            background: #f5f5f5;
        }

        /* toc entry, level 1 indentation */
        .toc_link[data-level="1"] {
            margin-left: 0;
        }

        /* toc entry, level 2 indentation */
        .toc_link[data-level="2"] {
            margin-left: 20px;
        }

        /* toc entry, level 3 indentation */
        .toc_link[data-level="3"] {
            margin-left: 40px;
        }

        /* toc entry, level 4 indentation */
        .toc_link[data-level="4"] {
            margin-left: 60px;
        }

        /* toc entry bullets */
        #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
            position: absolute;
            left: -15px;
            top: -1px;
            font-size: 1.5em;
        }

        /* toc entry, level 2 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
            content: "\2022";
        }

        /* toc entry, level 3 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
            content: "\25AB";
        }

        /* toc entry, level 4 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
            content: "-";
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* push <body> ("page") element down to make room for toc icon */
        .toc_body_nudge {
            padding-top: 60px;
        }

        /* toc icon when panel closed and not hovered */
        #toc_panel[data-open="false"]:not(:hover) {
            background: rgba(255, 255, 255, 0.75);
        }
    }

    /* always hide toc panel on print */
    @media only print {
        #toc_panel {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* lightbox plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* regular <img> in document when hovered */
        .lightbox_document_img:hover {
            cursor: pointer;
        }

        .body_no_scroll {
            overflow: hidden !important;
        }

        /* screen overlay */
        #lightbox_overlay {
            display: flex;
            flex-direction: column;
            position: fixed;
            left: 0;
            top: 0;
            right: 0;
            bottom: 0;
            background: rgba(0, 0, 0, 0.75);
            z-index: 3;
        }

        /* middle area containing lightbox image */
        #lightbox_image_container {
            flex-grow: 1;
            display: flex;
            justify-content: center;
            align-items: center;
            overflow: hidden;
            position: relative;
            padding: 20px;
        }

        /* bottom area containing caption */
        #lightbox_bottom_container {
            display: flex;
            justify-content: center;
            align-items: center;
            height: 100px;
            min-height: 100px;
            max-height: 100px;
            background: rgba(0, 0, 0, 0.5);
        }

        /* image number info text box */
        #lightbox_number_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            left: 2px;
            top: 0;
            z-index: 4;
        }

        /* zoom info text box */
        #lightbox_zoom_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            right: 2px;
            top: 0;
            z-index: 4;
        }

        /* copy of image caption */
        #lightbox_caption {
            box-sizing: border-box;
            display: inline-block;
            width: 100%;
            max-height: 100%;
            padding: 10px 0;
            text-align: center;
            overflow-y: auto;
            color: #ffffff;
        }

        /* navigation previous/next button */
        .lightbox_button {
            width: 100px;
            height: 100%;
            min-width: 100px;
            min-height: 100%;
            color: #ffffff;
        }

        /* navigation previous/next button when hovered */
        .lightbox_button:hover {
            background: none !important;
        }

        /* navigation button icon */
        .lightbox_button > svg {
            height: 25px;
        }

        /* figure auto-number */
        #lightbox_caption > span:first-of-type {
            font-weight: bold;
            margin-right: 5px;
        }

        /* lightbox image when hovered */
        #lightbox_img:hover {
            cursor: grab;
        }

        /* lightbox image when grabbed */
        #lightbox_img:active {
            cursor: grabbing;
        }
    }

    /* when on screen < 480px wide */
    @media only screen and (max-width: 480px) {
        /* make navigation buttons skinnier on small screens to make more room for caption text */
        .lightbox_button {
            width: 50px;
            min-width: 50px;
        }
    }

    /* always hide lightbox on print */
    @media only print {
        #lightbox_overlay {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* hypothesis (annotations) plugin */
    /* -------------------------------------------------- */

    /* hypothesis activation button */
    #hypothesis_button {
        box-sizing: border-box;
        position: fixed;
        top: 0;
        right: 0;
        width: 60px;
        height: 60px;
        background: #ffffff;
        border-radius: 0;
        border-left: solid 1px #bdbdbd;
        border-bottom: solid 1px #bdbdbd;
        box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
        z-index: 2;
    }

    /* hypothesis button svg */
    #hypothesis_button > svg {
        position: relative;
        top: -4px;
    }

    /* hypothesis annotation count */
    #hypothesis_count {
        position: absolute;
        left: 0;
        right: 0;
        bottom: 5px;
    }

    /* side panel */
    .annotator-frame {
        width: 280px !important;
    }

    /* match highlight color to rest of theme */
    .annotator-highlights-always-on .annotator-hl {
        background-color: #ffeb3b !important;
    }

    /* match focused color to rest of theme */
    .annotator-hl.annotator-hl-focused {
        background-color: #ff8a65 !important;
    }

    /* match bucket bar color to rest of theme */
    .annotator-bucket-bar {
        background: #f5f5f5 !important;
    }

    /* always hide button, toolbar, and tooltip on print */
    @media only print {
        #hypothesis_button {
            display: none;
        }

        .annotator-frame {
            display: none !important;
        }

        hypothesis-adder {
            display: none !important;
        }
    }
</style>
<!-- anchors plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds an anchor next to each of a certain type
        // of element that provides a human-readable url to that specific
        // item/position in the document (eg "manuscript.html#abstract"). It
        // also makes it such that scrolling out of view of a target removes
        // its identifier from the url.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'anchors';

        // default plugin options
        const options = {
            // which types of elements to add anchors next to, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3, [id^="fig:"], [id^="tbl:"], [id^="eq:"]',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // add anchor to each element of specified types
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements)
                addAnchor(element);

            // attach scroll listener to window
            window.addEventListener('scroll', onScroll);
        }

        // when window is scrolled
        function onScroll() {
            // if url has hash and user has scrolled out of view of hash
            // target, remove hash from url
            const tolerance = 100;
            const target = getHashTarget();
            if (target) {
                if (
                    target.getBoundingClientRect().top >
                        window.innerHeight + tolerance ||
                    target.getBoundingClientRect().bottom < 0 - tolerance
                )
                    history.pushState(null, null, ' ');
            }
        }

        // add anchor to element
        function addAnchor(element) {
            let addTo; // element to add anchor button to

            // if figure or table, modify withId and addTo to get expected
            // elements
            if (element.id.indexOf('fig:') === 0) {
                addTo = element.querySelector('figcaption');
            } else if (element.id.indexOf('tbl:') === 0) {
                addTo = element.querySelector('caption');
            } else if (element.id.indexOf('eq:') === 0) {
                addTo = element.querySelector('.eqnos-number');
            }

            addTo = addTo || element;
            const id = element.id || null;

            // do not add anchor if element doesn't have assigned id.
            // id is generated by pandoc and is assumed to be unique and
            // human-readable
            if (!id)
                return;

            // create anchor button
            const anchor = document.createElement('a');
            anchor.innerHTML = document.querySelector('.icon_link').innerHTML;
            anchor.title = 'Link to this part of the document';
            anchor.classList.add('icon_button', 'anchor');
            anchor.dataset.ignore = 'true';
            anchor.href = '#' + id;
            addTo.appendChild(anchor);
        }

        // get element that is target of link or url hash
        function getHashTarget() {
            const hash = window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- link icon -->

<template class="icon_link">
    <!-- modified from: https://fontawesome.com/icons/link -->
    <svg width="16" height="16" viewBox="0 0 512 512">
        <path
            fill="currentColor"
            d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
        ></path>
    </svg>
</template>
<!-- accordion plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows sections of content under <h2> headings
        // to be collapsible.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'accordion';

        // default plugin options
        const options = {
            // whether to always start expanded ('false'), always start
            // collapsed ('true'), or start collapsed when screen small ('auto')
            startCollapsed: 'auto',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <h2> heading
            const headings = document.querySelectorAll('h2');
            for (const heading of headings) {
                addArrow(heading);

                // start expanded/collapsed based on option
                if (
                    options.startCollapsed === 'true' ||
                    (options.startCollapsed === 'auto' && isSmallScreen())
                )
                    collapseHeading(heading);
                else
                    expandHeading(heading);
            }

            // attach hash change listener to window
            window.addEventListener('hashchange', onHashChange);
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                goToElement(target);
        }

        // add arrow to heading
        function addArrow(heading) {
            // add arrow button
            const arrow = document.createElement('button');
            arrow.innerHTML = document.querySelector(
                '.icon_angle_down'
            ).innerHTML;
            arrow.classList.add('icon_button', 'accordion_arrow');
            heading.insertBefore(arrow, heading.firstChild);

            // attach click listener to heading and button
            heading.addEventListener('click', onHeadingClick);
            arrow.addEventListener('click', onArrowClick);
        }

        // determine if on mobile-like device with small screen
        function isSmallScreen() {
            return Math.min(window.innerWidth, window.innerHeight) < 480;
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get element that is target of hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // when <h2> heading is clicked
        function onHeadingClick(event) {
            // only collapse if <h2> itself is target of click (eg, user did
            // not click on anchor within <h2>)
            if (event.target === this)
                toggleCollapse(this);
        }

        // when arrow button is clicked
        function onArrowClick() {
            toggleCollapse(this.parentNode);
        }

        // collapse section if expanded, expand if collapsed
        function toggleCollapse(heading) {
            if (heading.dataset.collapsed === 'false')
                collapseHeading(heading);
            else
                expandHeading(heading);
        }

        // elements to exclude from collapse, such as table of contents panel,
        // hypothesis panel, etc
        const exclude = '#toc_panel, div.annotator-frame, #lightbox_overlay';

        // collapse section
        function collapseHeading(heading) {
            heading.setAttribute('data-collapsed', 'true');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'true');
        }

        // expand section
        function expandHeading(heading) {
            heading.setAttribute('data-collapsed', 'false');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'false');
        }

        // get list of elements between this <h2> and next <h2> or <h1>
        // ("children" of the <h2> section)
        function getChildren(heading) {
            return nextUntil(heading, 'h2, h1', exclude);
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get list of elements after a start element up to element matching
        // query
        function nextUntil(element, query, exclude) {
            const elements = [];
            while (element = element.nextElementSibling, element) {
                if (element.matches(query))
                    break;
                if (!element.matches(exclude))
                    elements.push(element);
            }
            return elements;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
    <!-- modified from: https://fontawesome.com/icons/angle-down -->
    <svg width="16" height="16" viewBox="0 0 448 512">
        <path
            fill="currentColor"
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        ></path>
    </svg>
</template>
<!-- tooltips plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when the user hovers or
        // focuses a link to a citation or figure, a tooltip appears with a
        // preview of the reference content, along with arrows to navigate
        // between instances of the same reference in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tooltips';

        // default plugin options
        const options = {
            // whether user must click off to close tooltip instead of just
            // un-hovering
            clickClose: 'false',
            // delay (in ms) between opening and closing tooltip
            delay: '100',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach hover and focus listeners to link
                link.addEventListener('mouseover', onLinkHover);
                link.addEventListener('mouseleave', onLinkUnhover);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('touchend', onLinkTouch);
            }

            // attach mouse, key, and resize listeners to window
            window.addEventListener('mousedown', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('keyup', onKeyUp);
            window.addEventListener('resize', onResize);
        }

        // when link is hovered
        function onLinkHover() {
            // function to open tooltip
            const delayOpenTooltip = function() {
                openTooltip(this);
            }.bind(this);

            // run open function after delay
            this.openTooltipTimer = window.setTimeout(
                delayOpenTooltip,
                options.delay
            );
        }

        // when mouse leaves link
        function onLinkUnhover() {
            // cancel opening tooltip
            window.clearTimeout(this.openTooltipTimer);

            // don't close on unhover if option specifies
            if (options.clickClose === 'true')
                return;

            // function to close tooltip
            const delayCloseTooltip = function() {
                // if tooltip open and if mouse isn't over tooltip, close
                const tooltip = document.getElementById('tooltip');
                if (tooltip && !tooltip.matches(':hover'))
                    closeTooltip();
            };

            // run close function after delay
            this.closeTooltipTimer = window.setTimeout(
                delayCloseTooltip,
                options.delay
            );
        }

        // when link is focused (tabbed to)
        function onLinkFocus(event) {
            openTooltip(this);
        }

        // when link is touched on touch screen
        function onLinkTouch(event) {
            // attempt to force hover state on first tap always, and trigger
            // regular link click (and navigation) on second tap
            if (event.target === document.activeElement)
                event.target.click();
            else {
                document.activeElement.blur();
                event.target.focus();
            }
            if (event.cancelable)
                event.preventDefault();
            event.stopPropagation();
            return false;
        }

        // when mouse is clicked anywhere in window
        function onClick(event) {
            closeTooltip();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'tooltip_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'tooltip_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeTooltip();
                    break;
            }
        }

        // when window is resized or zoomed
        function onResize() {
            closeTooltip();
        }

        // get all links of types we wish to handle
        function getLinks() {
            const queries = [];
            // exclude buttons, anchor links, toc links, etc
            const exclude =
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            queries.push('a[href^="#ref-"]' + exclude); // citation links
            queries.push('a[href^="#fig:"]' + exclude); // figure links
            const query = queries.join(', ');
            return document.querySelectorAll(query);
        }

        // get links with same target, get index of link in set, get total
        // same links
        function getSameLinks(link) {
            const sameLinks = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    sameLinks.push(otherLink);
            }

            return {
                elements: sameLinks,
                index: sameLinks.indexOf(link),
                total: sameLinks.length
            };
        }

        // open tooltip
        function openTooltip(link) {
            // delete tooltip if it exists, start fresh
            closeTooltip();

            // make tooltip element
            const tooltip = makeTooltip(link);

            // if source couldn't be found and tooltip not made, exit
            if (!tooltip)
                return;

            // make navbar elements
            const navBar = makeNavBar(link);
            if (navBar)
                tooltip.firstElementChild.appendChild(navBar);

            // attach tooltip to page
            document.body.appendChild(tooltip);

            // position tooltip
            const position = function() {
                positionTooltip(link);
            };
            position();

            // if tooltip contains images, position again after they've loaded
            const imgs = tooltip.querySelectorAll('img');
            for (const img of imgs)
                img.addEventListener('load', position);
        }

        // close (delete) tooltip
        function closeTooltip() {
            const tooltip = document.getElementById('tooltip');
            if (tooltip)
                tooltip.remove();
        }

        // make tooltip
        function makeTooltip(link) {
            // get target element that link points to
            const source = getSource(link);

            // if source can't be found, exit
            if (!source)
                return;

            // create new tooltip
            const tooltip = document.createElement('div');
            tooltip.id = 'tooltip';
            const tooltipContent = document.createElement('div');
            tooltipContent.id = 'tooltip_content';
            tooltip.appendChild(tooltipContent);

            // make copy of source node and put in tooltip
            const sourceCopy = makeCopy(source);
            tooltipContent.appendChild(sourceCopy);

            // attach mouse event listeners
            tooltip.addEventListener('click', onTooltipClick);
            tooltip.addEventListener('mousedown', onTooltipClick);
            tooltip.addEventListener('touchstart', onTooltipClick);
            tooltip.addEventListener('mouseleave', onTooltipUnhover);

            // (for interaction with lightbox plugin)
            // transfer click on tooltip copied img to original img
            const sourceImg = source.querySelector('img');
            const sourceCopyImg = sourceCopy.querySelector('img');
            if (sourceImg && sourceCopyImg) {
                const clickImg = function() {
                    sourceImg.click();
                    closeTooltip();
                };
                sourceCopyImg.addEventListener('click', clickImg);
            }

            return tooltip;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // when tooltip is clicked
        function onTooltipClick(event) {
            // when user clicks on tooltip, stop click from transferring
            // outside of tooltip (eg, click off to close tooltip, or eg click
            // off to unhighlight same refs)
            event.stopPropagation();
        }

        // when tooltip is unhovered
        function onTooltipUnhover(event) {
            if (options.clickClose === 'true')
                return;

            // make sure new mouse/touch/focus no longer over tooltip or any
            // element within it
            const tooltip = document.getElementById('tooltip');
            if (!tooltip)
                return;
            if (this.contains(event.relatedTarget))
                return;

            closeTooltip();
        }

        // make nav bar to go betwen prev/next instances of same reference
        function makeNavBar(link) {
            // find other links to the same source
            const sameLinks = getSameLinks(link);

            // don't show nav bar when singular reference
            if (sameLinks.total <= 1)
                return;

            // find prev/next links with same target
            const prevLink = getPrevLink(link, sameLinks);
            const nextLink = getNextLink(link, sameLinks);

            // create nav bar
            const navBar = document.createElement('div');
            navBar.id = 'tooltip_nav_bar';
            const text = sameLinks.index + 1 + ' of ' + sameLinks.total;

            // create nav bar prev/next buttons
            const prevButton = document.createElement('button');
            const nextButton = document.createElement('button');
            prevButton.id = 'tooltip_prev_button';
            nextButton.id = 'tooltip_next_button';
            prevButton.title =
                'Jump to the previous occurence of this item in the document [←]';
            nextButton.title =
                'Jump to the next occurence of this item in the document [→]';
            prevButton.classList.add('icon_button');
            nextButton.classList.add('icon_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;
            navBar.appendChild(prevButton);
            navBar.appendChild(document.createTextNode(text));
            navBar.appendChild(nextButton);

            // attach click listeners to buttons
            prevButton.addEventListener('click', function() {
                onPrevNextClick(link, prevLink);
            });
            nextButton.addEventListener('click', function() {
                onPrevNextClick(link, nextLink);
            });

            return navBar;
        }

        // get previous link with same target
        function getPrevLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if < 1
            let index;
            if (sameLinks.index - 1 >= 0)
                index = sameLinks.index - 1;
            else
                index = sameLinks.total - 1;
            return sameLinks.elements[index];
        }

        // get next link with same target
        function getNextLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if > total
            let index;
            if (sameLinks.index + 1 <= sameLinks.total - 1)
                index = sameLinks.index + 1;
            else
                index = 0;
            return sameLinks.elements[index];
        }

        // get element that is target of link or url hash
        function getSource(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if ref or figure, modify target to get expected element
            if (id.indexOf('ref-') === 0)
                target = target.querySelector('p');
            else if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');

            return target;
        }

        // when prev/next arrow button is clicked
        function onPrevNextClick(link, prevNextLink) {
            if (link && prevNextLink)
                goToElement(prevNextLink, window.innerHeight * 0.5);
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // determine position to place tooltip based on link position in
        // viewport and tooltip size
        function positionTooltip(link, left, top) {
            const tooltipElement = document.getElementById('tooltip');
            if (!tooltipElement)
                return;

            // get convenient vars for position/dimensions of
            // link/tooltip/page/view
            link = getRectInPage(link);
            const tooltip = getRectInPage(tooltipElement);
            const view = getRectInPage();

            // horizontal positioning
            if (left)
                // use explicit value
                left = left;
            else if (link.left + tooltip.width < view.right)
                // fit tooltip to right of link
                left = link.left;
            else if (link.right - tooltip.width > view.left)
                // fit tooltip to left of link
                left = link.right - tooltip.width;
            // center tooltip in view
            else
                left = (view.right - view.left) / 2 - tooltip.width / 2;

            // vertical positioning
            if (top)
                // use explicit value
                top = top;
            else if (link.top - tooltip.height > view.top)
                // fit tooltip above link
                top = link.top - tooltip.height;
            else if (link.bottom + tooltip.height < view.bottom)
                // fit tooltip below link
                top = link.bottom;
            else {
                // center tooltip in view
                top = view.top + view.height / 2 - tooltip.height / 2;
                // nudge off of link to left/right if possible
                if (link.right + tooltip.width < view.right)
                    left = link.right;
                else if (link.left - tooltip.width > view.left)
                    left = link.left - tooltip.width;
            }

            tooltipElement.style.left = left + 'px';
            tooltipElement.style.top = top + 'px';
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get position of element relative to page
        function getRectInPage(element) {
            const rect = getRectInView(element);
            const body = getRectInView(document.body);

            const newRect = {};
            newRect.left = rect.left - body.left;
            newRect.top = rect.top - body.top;
            newRect.right = rect.right - body.left;
            newRect.bottom = rect.bottom - body.top;
            newRect.width = rect.width;
            newRect.height = rect.height;

            return newRect;
        }

        // (for interaction with accordion plugin)
        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // (for interaction with accordion plugin)
        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- jump to first plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds a button next to each reference entry,
        // figure, and table that jumps the page to the first occurrence of a
        // link to that item in the manuscript.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'jumpToFirst';

        // default plugin options
        const options = {
            // whether to add buttons next to reference entries
            references: 'true',
            // whether to add buttons next to figures
            figures: 'true',
            // whether to add buttons next to tables
            tables: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            if (options.references !== 'false')
                makeReferenceButtons();
            if (options.figures !== 'false')
                makeFigureButtons();
            if (options.tables !== 'false')
                makeTableButtons();
        }

        // when jump button clicked
        function onButtonClick() {
            const first = getFirstOccurrence(this.dataset.id);
            if (!first)
                return;

            // update url hash so navigating "back" in history will return
            // user to jump button
            window.location.hash = this.dataset.id;
            // scroll to link
            window.setTimeout(function() {
                goToElement(first, window.innerHeight * 0.5);
            }, 0);
        }

        // get first occurence of link to item in document
        function getFirstOccurrence(id) {
            let query = 'a';
            query += '[href="#' + id + '"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelector(query);
        }

        // add button next to each reference entry
        function makeReferenceButtons() {
            const references = document.querySelectorAll('div[id^="ref-"]');
            for (const reference of references) {
                // get reference id and element to add button to
                const id = reference.id;
                const container = reference.firstElementChild;
                const first = getFirstOccurrence(id);

                // if can't find link to reference, ignore
                if (!first)
                    continue;

                // make jump button
                let button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this reference in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.innerHTML = button.outerHTML + container.innerHTML;
                button = container.firstElementChild;
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeFigureButtons() {
            const figures = document.querySelectorAll('[id^="fig:"]');
            for (const figure of figures) {
                // get figure id and element to add button to
                const id = figure.id;
                const container = figure.querySelector('figcaption') || figure;
                const first = getFirstOccurrence(id);

                // if can't find link to figure, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this figure in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeTableButtons() {
            const tables = document.querySelectorAll('[id^="tbl:"]');
            for (const table of tables) {
                // get ref id and element to add button to
                const id = table.id;
                const container = table.querySelector('caption') || table;
                const first = getFirstOccurrence(id);

                // if can't find link to table, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this table in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
    <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
    <svg width="16" height="16" viewBox="0 0 320 512">
        <path
            fill="currentColor"
            d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
        ></path>
    </svg>
</template>
<!-- link highlight plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user hovers or
        // focuses a link, other links that have the same target will be
        // highlighted. It also makes it such that when clicking a link, the
        // target of the link (eg reference, figure, table) is briefly
        // highlighted.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'linkHighlight';

        // default plugin options
        const options = {
            // whether to also highlight links that go to external urls
            externalLinks: 'false',
            // whether user must click off to unhighlight instead of just
            // un-hovering
            clickUnhighlight: 'false',
            // whether to also highlight links that are unique
            highlightUnique: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach mouse and focus listeners to link
                link.addEventListener('mouseenter', onLinkFocus);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('mouseleave', onLinkUnhover);
            }

            // attach click and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('hashchange', onHashChange);

            // run hash change on window load in case user has navigated
            // directly to hash
            onHashChange();
        }

        // when link is focused (tabbed to) or hovered
        function onLinkFocus() {
            highlight(this);
        }

        // when link is unhovered
        function onLinkUnhover() {
            if (options.clickUnhighlight !== 'true')
                unhighlightAll();
        }

        // when the mouse is clicked anywhere in window
        function onClick(event) {
            unhighlightAll();
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                glowElement(target);
        }

        // get element that is target of link or url hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            return target;
        }

        // start glow sequence on an element
        function glowElement(element) {
            const startGlow = function() {
                onGlowEnd();
                element.dataset.glow = 'true';
                element.addEventListener('animationend', onGlowEnd);
            };
            const onGlowEnd = function() {
                element.removeAttribute('data-glow');
                element.removeEventListener('animationend', onGlowEnd);
            };
            startGlow();
        }

        // highlight link and all others with same target
        function highlight(link) {
            // force unhighlight all to start fresh
            unhighlightAll();

            // get links with same target
            if (!link)
                return;
            const sameLinks = getSameLinks(link);

            // if link unique and option is off, exit and don't highlight
            if (sameLinks.length <= 1 && options.highlightUnique !== 'true')
                return;

            // highlight all same links, and "select" (special highlight) this
            // one
            for (const sameLink of sameLinks) {
                if (sameLink === link)
                    sameLink.setAttribute('data-selected', 'true');
                else
                    sameLink.setAttribute('data-highlighted', 'true');
            }
        }

        // unhighlight all links
        function unhighlightAll() {
            const links = getLinks();
            for (const link of links) {
                link.setAttribute('data-selected', 'false');
                link.setAttribute('data-highlighted', 'false');
            }
        }

        // get links with same target
        function getSameLinks(link) {
            const results = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    results.push(otherLink);
            }
            return results;
        }

        // get all links of types we wish to handle
        function getLinks() {
            let query = 'a';
            if (options.externalLinks !== 'true')
                query += '[href^="#"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelectorAll(query);
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- table of contents plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin provides a "table of contents" (toc) panel on
        // the side of the document that allows the user to conveniently
        // navigate between sections of the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tableOfContents';

        // default plugin options
        const options = {
            // which types of elements to add links for, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3',
            // whether default behavior is to be closed ('false'), open
            // ('true'), or only open when screen wide enough to fit panel
            // ('auto'). note: still always starts closed when page loads.
            open: 'auto',
            // if list item is more than this many characters, text will be
            // truncated
            charLimit: '50',
            // whether or not to show bullets next to each toc item
            bullets: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // make toc panel and populate with entries (links to document
            // sections)
            const panel = makePanel();
            if (!panel)
                return;
            makeEntries(panel);
            document.body.insertBefore(panel, document.body.firstChild);

            closePanel();

            // attach click, scroll, and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('scroll', onScroll);
            window.addEventListener('hashchange', onScroll);
            window.addEventListener('keyup', onKeyUp);
            onScroll();

            // add class to push document body down out of way of toc button
            document.body.classList.add('toc_body_nudge');
        }

        // determine if screen wide enough to fit toc panel
        function isSmallScreen() {
            // in default theme:
            // 816px = 8.5in = width of "page" (<body>) element
            // 260px = min width of toc panel (*2 for both sides of <body>)
            return window.innerWidth < 816 + 260 * 2;
        }

        // open/close panel based on option and screen size
        function openOrClosePanel() {
            if (
                options.open === 'true' ||
                (options.open === 'auto' && !isSmallScreen())
            )
                openPanel();
            else
                closePanel();
        }

        // when mouse is clicked anywhere in window
        function onClick() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                openOrClosePanel();
        }

        // when window is scrolled or hash changed
        function onScroll() {
            highlightViewed();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            // close on esc
            if (event.key === 'Escape')
                closePanel();
        }

        // find entry of currently viewed document section in toc and highlight
        function highlightViewed() {
            const firstId = getFirstInView(options.typesQuery);

            // get toc entries (links), unhighlight all, then highlight viewed
            const list = document.getElementById('toc_list');
            if (!firstId || !list)
                return;
            const links = list.querySelectorAll('a');
            for (const link of links)
                link.dataset.viewing = 'false';
            const link = list.querySelector('a[href="#' + firstId + '"]');
            if (!link)
                return;
            link.dataset.viewing = 'true';
        }

        // get first or previous toc listed element in top half of view
        function getFirstInView(query) {
            // get all elements matching query and with id
            const elements = document.querySelectorAll(query);
            const elementsWithIds = [];
            for (const element of elements) {
                if (element.id)
                    elementsWithIds.push(element);
            }


            // get first or previous element in top half of view
            for (let i = 0; i < elementsWithIds.length; i++) {
                const element = elementsWithIds[i];
                const prevElement = elementsWithIds[Math.max(0, i - 1)];
                if (element.getBoundingClientRect().top >= 0) {
                    if (
                        element.getBoundingClientRect().top <
                        window.innerHeight / 2
                    )
                        return element.id;
                    else
                        return prevElement.id;
                }
            }
        }

        // make panel
        function makePanel() {
            // create panel
            const panel = document.createElement('div');
            panel.id = 'toc_panel';
            if (options.bullets === 'true')
                panel.dataset.bullets = 'true';

            // create header
            const header = document.createElement('div');
            header.id = 'toc_header';

            // create toc button
            const button = document.createElement('button');
            button.id = 'toc_button';
            button.innerHTML = document.querySelector('.icon_th_list').innerHTML;
            button.title = 'Table of Contents';
            button.classList.add('icon_button');

            // create header text
            const text = document.createElement('h3');
            text.innerHTML = 'View Table of Contents';

            // create container for toc list
            const list = document.createElement('div');
            list.id = 'toc_list';

            // attach click listeners
            panel.addEventListener('click', onPanelClick);
            header.addEventListener('click', onHeaderClick);
            button.addEventListener('click', onButtonClick);

            // attach elements
            header.appendChild(button);
            header.appendChild(text);
            panel.appendChild(header);
            panel.appendChild(list);

            return panel;
        }

        // create toc entries (links) to each element of the specified types
        function makeEntries(panel) {
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements) {
                // do not add link if element doesn't have assigned id
                if (!element.id)
                    continue;

                // create link/list item
                const link = document.createElement('a');
                link.classList.add('toc_link');
                switch (element.tagName.toLowerCase()) {
                    case 'h1':
                        link.dataset.level = '1';
                        break;
                    case 'h2':
                        link.dataset.level = '2';
                        break;
                    case 'h3':
                        link.dataset.level = '3';
                        break;
                    case 'h4':
                        link.dataset.level = '4';
                        break;
                }
                link.title = element.innerText;
                let text = element.innerText;
                if (text.length > options.charLimit)
                    text = text.slice(0, options.charLimit) + '...';
                link.innerHTML = text;
                link.href = '#' + element.id;
                link.addEventListener('click', onLinkClick);

                // attach link
                panel.querySelector('#toc_list').appendChild(link);
            }
        }

        // when panel is clicked
        function onPanelClick(event) {
            // stop click from propagating to window/document and closing panel
            event.stopPropagation();
        }

        // when header itself is clicked
        function onHeaderClick(event) {
            togglePanel();
        }

        // when button is clicked
        function onButtonClick(event) {
            togglePanel();
            // stop header underneath button from also being clicked
            event.stopPropagation();
        }

        // when link is clicked
        function onLinkClick() {
            openOrClosePanel();
        }

        // open panel if closed, close if opened
        function togglePanel() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                closePanel();
            else
                openPanel();
        }

        // open panel
        function openPanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'true';
        }

        // close panel
        function closePanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'false';
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- th list icon -->

<template class="icon_th_list">
    <!-- modified from: https://fontawesome.com/icons/th-list -->
    <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
        <path
            fill="currentColor"
            d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            tabindex="-1"
        ></path>
    </svg>
</template>
<!-- lightbox plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user clicks on an
        // image, the image fills the screen and the user can pan/drag/zoom
        // the image and navigate between other images in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'lightbox';

        // default plugin options
        const options = {
            // list of possible zoom/scale factors
            zoomSteps:
                '0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1,' +
                '1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8',
            // whether to fit image to view ('fit'), display at 100% and shrink
            // if necessary ('shrink'), or always display at 100% ('100')
            defaultZoom: 'fit',
            // whether to zoom in/out toward center of view ('true') or mouse
            // ('false')
            centerZoom: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <img> element
            const imgs = document.querySelectorAll('figure > img');
            let count = 1;
            for (const img of imgs) {
                img.classList.add('lightbox_document_img');
                img.dataset.number = count;
                img.dataset.total = imgs.length;
                img.addEventListener('click', openLightbox);
                count++;
            }

            // attach mouse and key listeners to window
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('keyup', onKeyUp);
        }

        // when mouse is moved anywhere in window
        function onWindowMouseMove(event) {
            window.mouseX = event.clientX;
            window.mouseY = event.clientY;
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'lightbox_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'lightbox_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeLightbox();
                    break;
            }
        }

        // open lightbox
        function openLightbox() {
            const lightbox = makeLightbox(this);
            if (!lightbox)
                return;

            blurBody(lightbox);
            document.body.appendChild(lightbox);
        }

        // make lightbox
        function makeLightbox(img) {
            // delete lightbox if it exists, start fresh
            closeLightbox();

            // create screen overlay containing lightbox
            const overlay = document.createElement('div');
            overlay.id = 'lightbox_overlay';

            // create image info boxes
            const numberInfo = document.createElement('div');
            const zoomInfo = document.createElement('div');
            numberInfo.id = 'lightbox_number_info';
            zoomInfo.id = 'lightbox_zoom_info';

            // create container for image
            const imageContainer = document.createElement('div');
            imageContainer.id = 'lightbox_image_container';
            const lightboxImg = makeLightboxImg(
                img,
                imageContainer,
                numberInfo,
                zoomInfo
            );
            imageContainer.appendChild(lightboxImg);

            // create bottom container for caption and navigation buttons
            const bottomContainer = document.createElement('div');
            bottomContainer.id = 'lightbox_bottom_container';
            const caption = makeCaption(img);
            const prevButton = makePrevButton(img);
            const nextButton = makeNextButton(img);
            bottomContainer.appendChild(prevButton);
            bottomContainer.appendChild(caption);
            bottomContainer.appendChild(nextButton);

            // attach top middle and bottom to overlay
            overlay.appendChild(numberInfo);
            overlay.appendChild(zoomInfo);
            overlay.appendChild(imageContainer);
            overlay.appendChild(bottomContainer);

            return overlay;
        }

        // make <img> object that is intuitively draggable and zoomable
        function makeLightboxImg(
            sourceImg,
            container,
            numberInfoBox,
            zoomInfoBox
        ) {
            // create copy of source <img>
            const img = sourceImg.cloneNode(true);
            img.classList.remove('lightbox_document_img');
            img.removeAttribute('id');
            img.removeAttribute('width');
            img.removeAttribute('height');
            img.style.position = 'unset';
            img.style.margin = '0';
            img.style.padding = '0';
            img.style.width = '';
            img.style.height = '';
            img.style.minWidth = '';
            img.style.minHeight = '';
            img.style.maxWidth = '';
            img.style.maxHeight = '';
            img.id = 'lightbox_img';

            // build sorted list of unique zoomSteps, always including a 100%
            let zoomSteps = [];
            const optionsZooms = options.zoomSteps.split(/[^0-9.]/);
            for (const optionZoom of optionsZooms) {
                const newZoom = parseFloat(optionZoom);
                if (newZoom && !zoomSteps.includes(newZoom))
                    zoomSteps.push(newZoom);
            }
            if (!zoomSteps.includes(1))
                zoomSteps.push(1);
            zoomSteps = zoomSteps.sort(function sortNumber(a, b) {
                return a - b;
            });

            // <img> object property variables
            let zoom = 1;
            let translateX = 0;
            let translateY = 0;
            let clickMouseX = undefined;
            let clickMouseY = undefined;
            let clickTranslateX = undefined;
            let clickTranslateY = undefined;

            updateNumberInfo();

            // update image numbers displayed in info box
            function updateNumberInfo() {
                numberInfoBox.innerHTML =
                    sourceImg.dataset.number + ' of ' + sourceImg.dataset.total;
            }

            // update zoom displayed in info box
            function updateZoomInfo() {
                let zoomInfo = zoom * 100;
                if (!Number.isInteger(zoomInfo))
                    zoomInfo = zoomInfo.toFixed(2);
                zoomInfoBox.innerHTML = zoomInfo + '%';
            }

            // move to closest zoom step above current zoom
            const zoomIn = function() {
                for (const zoomStep of zoomSteps) {
                    if (zoomStep > zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                updateTransform();
            };

            // move to closest zoom step above current zoom
            const zoomOut = function() {
                zoomSteps.reverse();
                for (const zoomStep of zoomSteps) {
                    if (zoomStep < zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                zoomSteps.reverse();

                updateTransform();
            };

            // update display of <img> based on scale/translate properties
            const updateTransform = function() {
                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                // get new width/height after scale
                const rect = img.getBoundingClientRect();
                // limit translate
                translateX = Math.max(translateX, -rect.width / 2);
                translateX = Math.min(translateX, rect.width / 2);
                translateY = Math.max(translateY, -rect.height / 2);
                translateY = Math.min(translateY, rect.height / 2);

                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                updateZoomInfo();
            };

            // fit <img> to container
            const fit = function() {
                // no x/y offset, 100% zoom by default
                translateX = 0;
                translateY = 0;
                zoom = 1;

                // widths of <img> and container
                const imgWidth = img.naturalWidth;
                const imgHeight = img.naturalHeight;
                const containerWidth = parseFloat(
                    window.getComputedStyle(container).width
                );
                const containerHeight = parseFloat(
                    window.getComputedStyle(container).height
                );

                // how much zooming is needed to fit <img> to container
                const xRatio = imgWidth / containerWidth;
                const yRatio = imgHeight / containerHeight;
                const maxRatio = Math.max(xRatio, yRatio);
                const newZoom = 1 / maxRatio;

                // fit <img> to container according to option
                if (options.defaultZoom === 'shrink') {
                    if (maxRatio > 1)
                        zoom = newZoom;
                } else if (options.defaultZoom === 'fit')
                    zoom = newZoom;

                updateTransform();
            };

            // when mouse wheel is rolled anywhere in container
            const onContainerWheel = function(event) {
                if (!event)
                    return;

                // let ctrl + mouse wheel to zoom behave as normal
                if (event.ctrlKey)
                    return;

                // prevent normal scroll behavior
                event.preventDefault();
                event.stopPropagation();

                // point around which to scale img
                const viewRect = container.getBoundingClientRect();
                const viewX = (viewRect.left + viewRect.right) / 2;
                const viewY = (viewRect.top + viewRect.bottom) / 2;
                const originX = options.centerZoom === 'true' ? viewX : mouseX;
                const originY = options.centerZoom === 'true' ? viewY : mouseY;

                // get point on image under origin
                const oldRect = img.getBoundingClientRect();
                const oldPercentX = (originX - oldRect.left) / oldRect.width;
                const oldPercentY = (originY - oldRect.top) / oldRect.height;

                // increment/decrement zoom
                if (event.deltaY < 0)
                    zoomIn();
                if (event.deltaY > 0)
                    zoomOut();

                // get offset between previous image point and origin
                const newRect = img.getBoundingClientRect();
                const offsetX =
                    originX - (newRect.left + newRect.width * oldPercentX);
                const offsetY =
                    originY - (newRect.top + newRect.height * oldPercentY);

                // translate image to keep image point under origin
                translateX += offsetX;
                translateY += offsetY;

                // perform translate
                updateTransform();
            };

            // when container is clicked
            function onContainerClick(event) {
                // if container itself is target of click, and not other
                // element above it
                if (event.target === this)
                    closeLightbox();
            }

            // when mouse button is pressed on image
            const onImageMouseDown = function(event) {
                // store original mouse position relative to image
                clickMouseX = window.mouseX;
                clickMouseY = window.mouseY;
                clickTranslateX = translateX;
                clickTranslateY = translateY;
                event.stopPropagation();
                event.preventDefault();
            };

            // when mouse button is released anywhere in window
            const onWindowMouseUp = function(event) {
                // reset original mouse position
                clickMouseX = undefined;
                clickMouseY = undefined;
                clickTranslateX = undefined;
                clickTranslateY = undefined;

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mouseup', onWindowMouseUp);
            };

            // when mouse is moved anywhere in window
            const onWindowMouseMove = function(event) {
                if (
                    clickMouseX === undefined ||
                    clickMouseY === undefined ||
                    clickTranslateX === undefined ||
                    clickTranslateY === undefined
                )
                    return;

                // offset image based on original and current mouse position
                translateX = clickTranslateX + window.mouseX - clickMouseX;
                translateY = clickTranslateY + window.mouseY - clickMouseY;
                updateTransform();
                event.preventDefault();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mousemove', onWindowMouseMove);
            };

            // when window is resized
            const onWindowResize = function(event) {
                fit();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('resize', onWindowResize);
            };

            // attach the necessary event listeners
            img.addEventListener('dblclick', fit);
            img.addEventListener('mousedown', onImageMouseDown);
            container.addEventListener('wheel', onContainerWheel);
            container.addEventListener('mousedown', onContainerClick);
            container.addEventListener('touchstart', onContainerClick);
            window.addEventListener('mouseup', onWindowMouseUp);
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('resize', onWindowResize);

            // run fit() after lightbox atttached to document and <img> Loaded
            // so needed container and img dimensions available
            img.addEventListener('load', fit);

            return img;
        }

        // make caption
        function makeCaption(img) {
            const caption = document.createElement('div');
            caption.id = 'lightbox_caption';
            const captionSource = img.nextElementSibling;
            if (captionSource.tagName.toLowerCase() === 'figcaption') {
                const captionCopy = makeCopy(captionSource);
                caption.innerHTML = captionCopy.innerHTML;
            }

            caption.addEventListener('touchstart', function(event) {
                event.stopPropagation();
            });

            return caption;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // make button to jump to previous image in document
        function makePrevButton(img) {
            const prevButton = document.createElement('button');
            prevButton.id = 'lightbox_prev_button';
            prevButton.title = 'Jump to the previous image in the document [←]';
            prevButton.classList.add('icon_button', 'lightbox_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;

            // attach click listeners to button
            prevButton.addEventListener('click', function() {
                getPrevImg(img).click();
            });

            return prevButton;
        }

        // make button to jump to next image in document
        function makeNextButton(img) {
            const nextButton = document.createElement('button');
            nextButton.id = 'lightbox_next_button';
            nextButton.title = 'Jump to the next image in the document [→]';
            nextButton.classList.add('icon_button', 'lightbox_button');
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;

            // attach click listeners to button
            nextButton.addEventListener('click', function() {
                getNextImg(img).click();
            });

            return nextButton;
        }

        // get previous image in document
        function getPrevImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if < 1
            if (index - 1 >= 0)
                index--;
            else
                index = imgs.length - 1;
            return imgs[index];
        }

        // get next image in document
        function getNextImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if > total
            if (index + 1 <= imgs.length - 1)
                index++;
            else
                index = 0;
            return imgs[index];
        }

        // close lightbox
        function closeLightbox() {
            focusBody();

            const lightbox = document.getElementById('lightbox_overlay');
            if (lightbox)
                lightbox.remove();
        }

        // make all elements behind lightbox non-focusable
        function blurBody(overlay) {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.tabIndex = -1;
            document.body.classList.add('body_no_scroll');
        }

        // make all elements focusable again
        function focusBody() {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.removeAttribute('tabIndex');
            document.body.classList.remove('body_no_scroll');
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- attributes plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows arbitrary HTML attributes to be attached
        // to (almost) any element. Place an HTML comment inside or next to the
        // desired element in the format <!-- $attribute="value" -->

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'attributes';

        // default plugin options
        const options = {
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // get list of comments in document
            const comments = findComments();

            for(const comment of comments)
                if (comment.parentElement)
                    addAttributes(
                        comment.parentElement,
                        comment.nodeValue.trim()
                    );
        }

        // add html attributes to specified element based on string of 
        // html attributes and values
        function addAttributes(element, text) {
            // regex's for finding attribute/value pairs in the format of
            // attribute="value" or attribute='value
            const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
            const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

            // loop through attribute/value pairs
            let match;
            while(match = text.match(regex2) || text.match(regex1)) {
                // get attribute and value from regex capture groups
                let attribute = match[1];
                let value = match[2];

                // remove from string
                text = text.substring(match.index + match[0].length);

                if (!attribute || !value)
                    break;

                // set attribute of parent element
                try {
                    element.setAttribute(attribute, value);
                } catch(error) {
                    console.log(error);
                }

                // special case for colspan
                if (attribute === 'colspan')
                    removeTableCells(element, value);
            }
        }

        // get list of comment elements in document
        function findComments() {
            const comments = [];

            // iterate over comment nodes in document
            function acceptNode(node) {
                return NodeFilter.FILTER_ACCEPT;
            }
            const iterator = document.createNodeIterator(
                document.body,
                NodeFilter.SHOW_COMMENT,
                acceptNode
            );
            let node;
            while(node = iterator.nextNode())
                comments.push(node);

            return comments;
        }

        // remove certain number of cells after specified cell
        function removeTableCells(cell, number) {
            number = parseInt(number);
            if (!number)
                return;

            // remove elements
            for(; number > 1; number--) {
                if (cell.nextElementSibling)
                    cell.nextElementSibling.remove();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- math plugin configuration -->

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        "CommonHTML": { linebreaks: { automatic: true } },
        "HTML-CSS": { linebreaks: { automatic: true } },
        "SVG": { linebreaks: { automatic: true } },
        "fast-preview": { disabled: true }
    });
</script>

<!-- math plugin -->

<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML'>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'MathJax' allows the proper rendering of
    // math/equations written in LaTeX.

    // https://www.mathjax.org/
</script>
<!-- annotations plugin -->

<script>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'Hypothesis' allows public annotation of the
    // manuscript.

    // https://web.hypothes.is/

    // plugin configuration
    window.hypothesisConfig = function() {
        return {
            branding: {
                accentColor: '#2196f3',
                appBackgroundColor: '#f8f8f8',
                ctaBackgroundColor: '#f8f8f8',
                ctaTextColor: '#000000',
                selectionFontFamily: 'Open Sans, Helvetica, sans serif',
                annotationFontFamily: 'Open Sans, Helvetica, sans serif'
            }
        };
    };

    // hypothesis client script
    const embed = 'https://hypothes.is/embed.js';
    // hypothesis annotation count query url
    const query = 'https://api.hypothes.is/api/search?limit=0&url='

    
    // start script
    function start() {
        const button = makeButton();
        document.body.insertBefore(button, document.body.firstChild);
        insertCount(button);
    }

    // make button
    function makeButton() {
        // create button
        const button = document.createElement('button');
        button.id = 'hypothesis_button';
        button.innerHTML = document.querySelector('.icon_hypothesis').innerHTML;
        button.title = 'Hypothesis annotations';
        button.classList.add('icon_button');

        function onClick(event) {
            onButtonClick(event, button);
        }

        // attach click listeners
        button.addEventListener('click', onClick);

        return button;
    }

    // insert annotations count
    async function insertCount(button) {
        // get annotation count from Hypothesis based on url
        let count = '-';
        try {
            const canonical = document.querySelector('link[rel="canonical"]');
            const location = window.location;
            const url = encodeURIComponent((canonical || location).href);
            const response = await fetch(query + url);
            const json = await response.json();
            count = json.total || '-';
        } catch(error) {
            console.log(error);
        }
        
        // put count into button
        const counter = document.createElement('span');
        counter.id = 'hypothesis_count';
        counter.innerHTML = count;
        button.title = 'View ' + count + ' Hypothesis annotations';
        button.append(counter);
    }

    // when button is clicked
    function onButtonClick(event, button) {
        const script = document.createElement('script');
        script.src = embed;
        document.body.append(script);
        button.remove();
    }

    window.addEventListener('load', start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
    <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
    <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
        <path
            fill="currentColor"
            d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
            tabindex="-1"
        ></path>
    </svg>
</template>
<!-- analytics plugin -->

<!-- copy and paste code from Google Analytics or similar service here -->
</body>
</html>
